{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "044e373e",
   "metadata": {},
   "source": [
    "# GRN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "99d69bb9",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T11:19:40.430544Z",
     "start_time": "2022-05-17T10:51:36.666562Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-05-17 10:51:46,975 - pyscenic.cli.pyscenic - INFO - Loading expression matrix.\n",
      "\n",
      "2022-05-17 10:52:12,037 - pyscenic.cli.pyscenic - INFO - Inferring regulatory networks.\n",
      "preparing dask client\n",
      "parsing input\n",
      "creating dask graph\n",
      "32 partitions\n",
      "computing dask graph\n",
      "not shutting down client, client was created externally\n",
      "finished\n",
      "distributed.worker - WARNING - Heartbeat to scheduler failed\n",
      "\n",
      "2022-05-17 11:19:31,410 - pyscenic.cli.pyscenic - INFO - Writing results to file.\n"
     ]
    }
   ],
   "source": [
    "!docker run -it --rm \\\n",
    "    -v /home/csj/mywork/ilc/220517-pyscenic-grn:/scenicdata \\\n",
    "    aertslab/pyscenic:0.11.2 pyscenic grn \\\n",
    "        --num_workers 32 \\\n",
    "        -o /scenicdata/expr_mat.adjacencies_abm.tsv \\\n",
    "        /scenicdata/expr_mat_abm.tsv \\\n",
    "        /scenicdata/mm_mgi_tfs.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "25278d35",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T11:20:47.591148Z",
     "start_time": "2022-05-17T11:20:45.916734Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "grn = pd.read_csv('expr_mat.adjacencies_abm.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "68229850",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T11:23:22.598676Z",
     "start_time": "2022-05-17T11:23:22.237434Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TF</th>\n",
       "      <th>target</th>\n",
       "      <th>importance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Gata3</td>\n",
       "      <td>Id2</td>\n",
       "      <td>9.210185e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Gata3</td>\n",
       "      <td>Irf8</td>\n",
       "      <td>8.698182e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>Gata3</td>\n",
       "      <td>Mef2c</td>\n",
       "      <td>5.955541e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>Gata3</td>\n",
       "      <td>Ighm</td>\n",
       "      <td>5.533995e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>Gata3</td>\n",
       "      <td>Clec12a</td>\n",
       "      <td>5.531352e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2710676</th>\n",
       "      <td>Gata3</td>\n",
       "      <td>D630029K05Rik</td>\n",
       "      <td>5.576024e-20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2710722</th>\n",
       "      <td>Gata3</td>\n",
       "      <td>Ank1</td>\n",
       "      <td>5.287670e-20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2710768</th>\n",
       "      <td>Gata3</td>\n",
       "      <td>A830008E24Rik</td>\n",
       "      <td>4.981548e-20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2711135</th>\n",
       "      <td>Gata3</td>\n",
       "      <td>1700007K13Rik</td>\n",
       "      <td>2.639301e-20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2711249</th>\n",
       "      <td>Gata3</td>\n",
       "      <td>Ptprn</td>\n",
       "      <td>1.926475e-20</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6672 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            TF         target    importance\n",
       "35       Gata3            Id2  9.210185e+01\n",
       "52       Gata3           Irf8  8.698182e+01\n",
       "261      Gata3          Mef2c  5.955541e+01\n",
       "341      Gata3           Ighm  5.533995e+01\n",
       "342      Gata3        Clec12a  5.531352e+01\n",
       "...        ...            ...           ...\n",
       "2710676  Gata3  D630029K05Rik  5.576024e-20\n",
       "2710722  Gata3           Ank1  5.287670e-20\n",
       "2710768  Gata3  A830008E24Rik  4.981548e-20\n",
       "2711135  Gata3  1700007K13Rik  2.639301e-20\n",
       "2711249  Gata3          Ptprn  1.926475e-20\n",
       "\n",
       "[6672 rows x 3 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grn[grn['TF']=='Gata3']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be5dca74",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8eb45724",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "37fedb26",
   "metadata": {},
   "source": [
    "# CTX"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "a1546ec9",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T16:12:31.778527Z",
     "start_time": "2022-05-17T15:22:08.167457Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-05-17 15:22:18,739 - pyscenic.cli.pyscenic - INFO - Creating modules.\n",
      "\n",
      "2022-05-17 15:22:20,583 - pyscenic.cli.pyscenic - INFO - Loading expression matrix.\n",
      "\n",
      "2022-05-17 15:22:44,849 - pyscenic.utils - INFO - Calculating Pearson correlations.\n",
      "\n",
      "2022-05-17 15:22:45,670 - pyscenic.utils - WARNING - Note on correlation calculation: the default behaviour for calculating the correlations has changed after pySCENIC verion 0.9.16. Previously, the default was to calculate the correlation between a TF and target gene using only cells with non-zero expression values (mask_dropouts=True). The current default is now to use all cells to match the behavior of the R verision of SCENIC. The original settings can be retained by setting 'rho_mask_dropouts=True' in the modules_from_adjacencies function, or '--mask_dropouts' from the CLI.\n",
      "\tDropout masking is currently set to [False].\n",
      "\n",
      "2022-05-17 15:23:43,171 - pyscenic.utils - INFO - Creating modules.\n",
      "\n",
      "2022-05-17 15:26:26,868 - pyscenic.cli.pyscenic - INFO - Loading databases.\n",
      "\n",
      "2022-05-17 15:26:26,868 - pyscenic.cli.pyscenic - INFO - Calculating regulons.\n",
      "[                                        ] | 0% Completed | 15.0s\n",
      "2022-05-17 15:26:44,878 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm9833 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 41.3s\n",
      "2022-05-17 15:27:11,174 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Snai1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 13min 57.8s\n",
      "2022-05-17 15:40:27,654 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Dmrta2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 26min 43.4s\n",
      "2022-05-17 15:53:13,333 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp964 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 26min 44.3s\n",
      "2022-05-17 15:53:14,211 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp970 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 26min 46.8s\n",
      "2022-05-17 15:53:16,695 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zkscan4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 26min 52.6s\n",
      "2022-05-17 15:53:22,413 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp558 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 29min 48.7s\n",
      "2022-05-17 15:56:18,604 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zkscan4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 32min  0.9s\n",
      "2022-05-17 15:58:30,750 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Pou2af1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 32min 45.7s\n",
      "2022-05-17 15:59:15,571 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Hdx could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 32min 54.6s\n",
      "2022-05-17 15:59:24,366 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Foxf1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 32min 58.5s\n",
      "2022-05-17 15:59:28,320 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Irx3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 33min  0.1s\n",
      "2022-05-17 15:59:29,996 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm10093 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 33min  5.1s\n",
      "2022-05-17 15:59:34,909 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp369 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 33min  7.4s\n",
      "2022-05-17 15:59:37,243 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp433 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 33min  8.7s\n",
      "2022-05-17 15:59:38,555 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm14443 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 1% Completed | 33min 23.1s\n",
      "2022-05-17 15:59:53,004 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm3854 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 1% Completed | 33min 23.2s\n",
      "2022-05-17 15:59:53,070 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Nuak1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 1% Completed | 33min 23.9s\n",
      "2022-05-17 15:59:53,780 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm9833 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 1% Completed | 33min 28.2s\n",
      "2022-05-17 15:59:58,133 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp551 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 1% Completed | 33min 29.3s\n",
      "2022-05-17 15:59:59,205 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Edn1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 2% Completed | 33min 37.4s\n",
      "2022-05-17 16:00:07,227 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Dmrta2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 2% Completed | 33min 45.1s\n",
      "2022-05-17 16:00:15,011 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zkscan4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 4% Completed | 33min 56.5s\n",
      "2022-05-17 16:00:26,376 - pyscenic.transform - WARNING - Less than 80% of the genes in E430018J23Rik could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 4% Completed | 33min 58.8s\n",
      "2022-05-17 16:00:28,684 - pyscenic.transform - WARNING - Less than 80% of the genes in Ebf1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 4% Completed | 33min 58.9s\n",
      "2022-05-17 16:00:28,818 - pyscenic.transform - WARNING - Less than 80% of the genes in Ebf4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 4% Completed | 34min  1.6s\n",
      "2022-05-17 16:00:31,589 - pyscenic.transform - WARNING - Less than 80% of the genes in 2610044O15Rik8 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 4% Completed | 34min  2.7s\n",
      "2022-05-17 16:00:32,653 - pyscenic.transform - WARNING - Less than 80% of the genes in 9130023H24Rik could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 4% Completed | 34min  3.7s\n",
      "2022-05-17 16:00:33,635 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Dmrta2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 4% Completed | 34min  4.0s\n",
      "2022-05-17 16:00:33,973 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm17655 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 4% Completed | 34min  4.4s\n",
      "2022-05-17 16:00:34,272 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm20939 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 4% Completed | 34min  4.7s\n",
      "2022-05-17 16:00:34,599 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm3055 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 4% Completed | 34min  4.8s\n",
      "2022-05-17 16:00:34,653 - pyscenic.transform - WARNING - Less than 80% of the genes in Agap2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 4% Completed | 34min  5.2s\n",
      "2022-05-17 16:00:35,076 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm38394 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 5% Completed | 34min 12.7s\n",
      "2022-05-17 16:00:42,600 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Tal1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 5% Completed | 34min 13.5s\n",
      "2022-05-17 16:00:43,337 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm6710 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 5% Completed | 34min 13.6s\n",
      "2022-05-17 16:00:43,473 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm9833 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 6% Completed | 34min 16.1s\n",
      "2022-05-17 16:00:46,011 - pyscenic.transform - WARNING - Less than 80% of the genes in Cd59b could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 6% Completed | 34min 16.5s\n",
      "2022-05-17 16:00:46,410 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp551 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 6% Completed | 34min 16.7s\n",
      "2022-05-17 16:00:46,623 - pyscenic.transform - WARNING - Less than 80% of the genes in Mnt could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 6% Completed | 34min 17.6s\n",
      "2022-05-17 16:00:47,554 - pyscenic.transform - WARNING - Less than 80% of the genes in Msi2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 7% Completed | 34min 18.7s\n",
      "2022-05-17 16:00:48,598 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp964 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 7% Completed | 34min 19.8s\n",
      "2022-05-17 16:00:49,702 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Cd59b could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 7% Completed | 34min 19.9s\n",
      "2022-05-17 16:00:49,822 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Cfl2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 7% Completed | 34min 20.3s\n",
      "2022-05-17 16:00:50,220 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Snai1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 7% Completed | 34min 20.4s\n",
      "2022-05-17 16:00:50,356 - pyscenic.transform - WARNING - Less than 80% of the genes in Msrb3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 7% Completed | 34min 20.8s\n",
      "2022-05-17 16:00:50,793 - pyscenic.transform - WARNING - Less than 80% of the genes in Phtf1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##                                      ] | 7% Completed | 34min 21.0s\n",
      "2022-05-17 16:00:50,929 - pyscenic.transform - WARNING - Less than 80% of the genes in Pick1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###                                     ] | 7% Completed | 34min 21.8s\n",
      "2022-05-17 16:00:51,711 - pyscenic.transform - WARNING - Less than 80% of the genes in Ascc1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###                                     ] | 7% Completed | 34min 22.6s\n",
      "2022-05-17 16:00:52,508 - pyscenic.transform - WARNING - Less than 80% of the genes in Grhl3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###                                     ] | 8% Completed | 34min 28.3s\n",
      "2022-05-17 16:00:58,239 - pyscenic.transform - WARNING - Less than 80% of the genes in Mterf1a could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###                                     ] | 8% Completed | 34min 29.0s\n",
      "2022-05-17 16:00:58,832 - pyscenic.transform - WARNING - Less than 80% of the genes in Mxd1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###                                     ] | 8% Completed | 34min 30.9s\n",
      "2022-05-17 16:01:00,768 - pyscenic.transform - WARNING - Less than 80% of the genes in Irf3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###                                     ] | 8% Completed | 34min 31.8s\n",
      "2022-05-17 16:01:01,769 - pyscenic.transform - WARNING - Less than 80% of the genes in Tcf7l2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###                                     ] | 9% Completed | 34min 32.4s\n",
      "2022-05-17 16:01:02,242 - pyscenic.transform - WARNING - Less than 80% of the genes in Gtpbp1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###                                     ] | 9% Completed | 34min 32.6s\n",
      "2022-05-17 16:01:02,516 - pyscenic.transform - WARNING - Less than 80% of the genes in Gzf1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###                                     ] | 9% Completed | 34min 32.7s\n",
      "2022-05-17 16:01:02,649 - pyscenic.transform - WARNING - Less than 80% of the genes in H1fx could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###                                     ] | 9% Completed | 34min 32.8s\n",
      "2022-05-17 16:01:02,692 - pyscenic.transform - WARNING - Less than 80% of the genes in Tet1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[###                                     ] | 9% Completed | 34min 34.2s\n",
      "2022-05-17 16:01:04,191 - pyscenic.transform - WARNING - Less than 80% of the genes in Zbtb42 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####                                    ] | 10% Completed | 34min 35.3s\n",
      "2022-05-17 16:01:05,199 - pyscenic.transform - WARNING - Less than 80% of the genes in Zbtb49 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####                                    ] | 10% Completed | 34min 35.5s\n",
      "2022-05-17 16:01:05,343 - pyscenic.transform - WARNING - Less than 80% of the genes in Zbtb5 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####                                    ] | 10% Completed | 34min 35.8s\n",
      "2022-05-17 16:01:05,644 - pyscenic.transform - WARNING - Less than 80% of the genes in Tfap2a could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####                                    ] | 10% Completed | 34min 37.4s\n",
      "2022-05-17 16:01:07,329 - pyscenic.transform - WARNING - Less than 80% of the genes in Atf4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####                                    ] | 10% Completed | 34min 38.2s\n",
      "2022-05-17 16:01:08,140 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp426 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####                                    ] | 10% Completed | 34min 38.4s\n",
      "2022-05-17 16:01:08,298 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp433 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:08,337 - pyscenic.transform - WARNING - Less than 80% of the genes in Pou2af1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####                                    ] | 11% Completed | 34min 39.5s\n",
      "2022-05-17 16:01:09,370 - pyscenic.transform - WARNING - Less than 80% of the genes in Scmh1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####                                    ] | 11% Completed | 34min 40.7s\n",
      "2022-05-17 16:01:10,601 - pyscenic.transform - WARNING - Less than 80% of the genes in Atf6 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####                                    ] | 11% Completed | 34min 40.8s\n",
      "2022-05-17 16:01:10,733 - pyscenic.transform - WARNING - Less than 80% of the genes in Atf6b could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####                                    ] | 11% Completed | 34min 41.2s\n",
      "2022-05-17 16:01:11,066 - pyscenic.transform - WARNING - Less than 80% of the genes in Atoh8 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####                                    ] | 11% Completed | 34min 41.3s\n",
      "2022-05-17 16:01:11,205 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp551 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####                                    ] | 11% Completed | 34min 41.6s\n",
      "2022-05-17 16:01:11,491 - pyscenic.transform - WARNING - Less than 80% of the genes in BC005561 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 45.6s\n",
      "2022-05-17 16:01:15,589 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp105 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 46.3s\n",
      "2022-05-17 16:01:16,299 - pyscenic.transform - WARNING - Less than 80% of the genes in Myog could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 46.7s\n",
      "2022-05-17 16:01:16,654 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp114 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 48.8s\n",
      "2022-05-17 16:01:18,653 - pyscenic.transform - WARNING - Less than 80% of the genes in Foxf1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 49.4s\n",
      "2022-05-17 16:01:19,332 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm14443 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 50.0s\n",
      "2022-05-17 16:01:19,891 - pyscenic.transform - WARNING - Less than 80% of the genes in Cfl2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 50.6s\n",
      "2022-05-17 16:01:20,529 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm3055 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 50.7s\n",
      "2022-05-17 16:01:20,564 - pyscenic.transform - WARNING - Less than 80% of the genes in Esrra could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 51.0s\n",
      "2022-05-17 16:01:20,882 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp592 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 51.1s\n",
      "2022-05-17 16:01:21,056 - pyscenic.transform - WARNING - Less than 80% of the genes in Hdx could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 51.2s\n",
      "2022-05-17 16:01:21,068 - pyscenic.transform - WARNING - Less than 80% of the genes in Bax could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 51.3s\n",
      "2022-05-17 16:01:21,206 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm3854 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 52.0s\n",
      "2022-05-17 16:01:21,922 - pyscenic.transform - WARNING - Less than 80% of the genes in Tfeb could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 52.2s\n",
      "2022-05-17 16:01:22,064 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm9833 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 52.5s\n",
      "2022-05-17 16:01:22,421 - pyscenic.transform - WARNING - Less than 80% of the genes in Tgif2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 12% Completed | 34min 53.0s\n",
      "2022-05-17 16:01:22,909 - pyscenic.transform - WARNING - Less than 80% of the genes in Arg2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 34min 55.7s\n",
      "2022-05-17 16:01:25,631 - pyscenic.transform - WARNING - Less than 80% of the genes in Nfatc1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:25,682 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp131 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[#####                                   ] | 13% Completed | 34min 56.0s\n",
      "2022-05-17 16:01:25,981 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp143 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 34min 56.3s\n",
      "2022-05-17 16:01:26,266 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp148 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 34min 56.5s\n",
      "2022-05-17 16:01:26,401 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Edn1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 34min 57.5s\n",
      "2022-05-17 16:01:27,458 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp90 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 34min 59.2s\n",
      "2022-05-17 16:01:29,092 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp941 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  0.4s\n",
      "2022-05-17 16:01:30,382 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp964 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  0.7s\n",
      "2022-05-17 16:01:30,614 - pyscenic.transform - WARNING - Less than 80% of the genes in Hic2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  1.2s\n",
      "2022-05-17 16:01:31,172 - pyscenic.transform - WARNING - Less than 80% of the genes in Smad2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  1.5s\n",
      "2022-05-17 16:01:31,405 - pyscenic.transform - WARNING - Less than 80% of the genes in Creb1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  2.2s\n",
      "2022-05-17 16:01:32,103 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfx could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  3.2s\n",
      "2022-05-17 16:01:33,119 - pyscenic.transform - WARNING - Less than 80% of the genes in Atf6 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  3.4s\n",
      "2022-05-17 16:01:33,379 - pyscenic.transform - WARNING - Less than 80% of the genes in Zkscan3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  3.6s\n",
      "2022-05-17 16:01:33,531 - pyscenic.transform - WARNING - Less than 80% of the genes in Thap11 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:33,564 - pyscenic.transform - WARNING - Less than 80% of the genes in Zkscan4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  3.8s\n",
      "2022-05-17 16:01:33,721 - pyscenic.transform - WARNING - Less than 80% of the genes in Zkscan5 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  3.9s\n",
      "2022-05-17 16:01:33,814 - pyscenic.transform - WARNING - Less than 80% of the genes in Atoh8 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  4.0s\n",
      "2022-05-17 16:01:33,967 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Foxf1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  4.1s\n",
      "2022-05-17 16:01:34,069 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp704 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  4.5s\n",
      "2022-05-17 16:01:34,395 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  4.8s\n",
      "2022-05-17 16:01:34,693 - pyscenic.transform - WARNING - Less than 80% of the genes in Hivep3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  5.0s\n",
      "2022-05-17 16:01:34,835 - pyscenic.transform - WARNING - Less than 80% of the genes in Dbx2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  5.2s\n",
      "2022-05-17 16:01:35,080 - pyscenic.transform - WARNING - Less than 80% of the genes in Tmem33 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  6.2s\n",
      "2022-05-17 16:01:36,069 - pyscenic.transform - WARNING - Less than 80% of the genes in Trim24 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  7.8s\n",
      "2022-05-17 16:01:37,643 - pyscenic.transform - WARNING - Less than 80% of the genes in Hdx could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  8.2s\n",
      "2022-05-17 16:01:38,019 - pyscenic.transform - WARNING - Less than 80% of the genes in Hes5 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  8.8s\n",
      "2022-05-17 16:01:38,701 - pyscenic.transform - WARNING - Less than 80% of the genes in Irx3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  9.3s\n",
      "2022-05-17 16:01:39,140 - pyscenic.transform - WARNING - Less than 80% of the genes in Ppp1r10 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:39,183 - pyscenic.transform - WARNING - Less than 80% of the genes in Jdp2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min  9.5s\n",
      "2022-05-17 16:01:39,398 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Irx3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min 10.2s\n",
      "2022-05-17 16:01:40,067 - pyscenic.transform - WARNING - Less than 80% of the genes in Prdm11 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min 10.7s\n",
      "2022-05-17 16:01:40,604 - pyscenic.transform - WARNING - Less than 80% of the genes in Prdm4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min 11.4s\n",
      "2022-05-17 16:01:41,227 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp729b could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[#####                                   ] | 13% Completed | 35min 11.7s\n",
      "2022-05-17 16:01:41,589 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp746 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min 12.7s\n",
      "2022-05-17 16:01:42,614 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp775 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min 13.2s\n",
      "2022-05-17 16:01:43,059 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp235 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed | 35min 13.5s\n",
      "2022-05-17 16:01:43,384 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp787 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 14% Completed | 35min 14.3s\n",
      "2022-05-17 16:01:44,224 - pyscenic.transform - WARNING - Less than 80% of the genes in Zscan12 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 14% Completed | 35min 15.5s\n",
      "2022-05-17 16:01:45,357 - pyscenic.transform - WARNING - Less than 80% of the genes in Zscan29 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 14% Completed | 35min 15.8s\n",
      "2022-05-17 16:01:45,674 - pyscenic.transform - WARNING - Less than 80% of the genes in Zxdc could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 14% Completed | 35min 16.1s\n",
      "2022-05-17 16:01:46,043 - pyscenic.transform - WARNING - Less than 80% of the genes in Hic2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 14% Completed | 35min 16.4s\n",
      "2022-05-17 16:01:46,271 - pyscenic.transform - WARNING - Less than 80% of the genes in 2610044O15Rik8 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 14% Completed | 35min 16.8s\n",
      "2022-05-17 16:01:46,655 - pyscenic.transform - WARNING - Less than 80% of the genes in Smap2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 14% Completed | 35min 17.3s\n",
      "2022-05-17 16:01:47,148 - pyscenic.transform - WARNING - Less than 80% of the genes in 9130023H24Rik could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[######                                  ] | 15% Completed | 35min 17.7s\n",
      "2022-05-17 16:01:47,586 - pyscenic.transform - WARNING - Less than 80% of the genes in AW146154 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[######                                  ] | 15% Completed | 35min 18.4s\n",
      "2022-05-17 16:01:48,263 - pyscenic.transform - WARNING - Less than 80% of the genes in Aco1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[######                                  ] | 16% Completed | 35min 22.4s\n",
      "2022-05-17 16:01:53,396 - pyscenic.transform - WARNING - Less than 80% of the genes in Hoxa6 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:54,457 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp296 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:54,541 - pyscenic.transform - WARNING - Less than 80% of the genes in Vps4b could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:54,693 - pyscenic.transform - WARNING - Less than 80% of the genes in Dmrta2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:54,755 - pyscenic.transform - WARNING - Less than 80% of the genes in Wdr83 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:55,455 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp319 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:55,492 - pyscenic.transform - WARNING - Less than 80% of the genes in Snai1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:55,587 - pyscenic.transform - WARNING - Less than 80% of the genes in Hoxa7 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:56,664 - pyscenic.transform - WARNING - Less than 80% of the genes in Dus3l could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:56,834 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp362 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:56,954 - pyscenic.transform - WARNING - Less than 80% of the genes in Wrnip1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:57,085 - pyscenic.transform - WARNING - Less than 80% of the genes in Rb1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:57,209 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Edn1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:57,545 - pyscenic.transform - WARNING - Less than 80% of the genes in Nf1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:57,909 - pyscenic.transform - WARNING - Less than 80% of the genes in Nfatc1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:58,042 - pyscenic.transform - WARNING - Less than 80% of the genes in Bhlha15 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:58,279 - pyscenic.transform - WARNING - Less than 80% of the genes in Snapc4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:58,358 - pyscenic.transform - WARNING - Less than 80% of the genes in Bhlhe41 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:59,318 - pyscenic.transform - WARNING - Less than 80% of the genes in Sox12 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:59,642 - pyscenic.transform - WARNING - Less than 80% of the genes in Nfyc could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:01:59,901 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp369 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:00,031 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp382 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:00,683 - pyscenic.transform - WARNING - Less than 80% of the genes in Nnt could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:02,808 - pyscenic.transform - WARNING - Less than 80% of the genes in Rbbp5 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:04,914 - pyscenic.transform - WARNING - Less than 80% of the genes in Nr2c2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-05-17 16:02:05,092 - pyscenic.transform - WARNING - Less than 80% of the genes in Cxxc1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:05,614 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Foxf1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:08,002 - pyscenic.transform - WARNING - Less than 80% of the genes in Hoxa2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:08,171 - pyscenic.transform - WARNING - Less than 80% of the genes in Hoxa3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:08,556 - pyscenic.transform - WARNING - Less than 80% of the genes in Hoxa6 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:08,802 - pyscenic.transform - WARNING - Less than 80% of the genes in Hoxa9 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:09,197 - pyscenic.transform - WARNING - Less than 80% of the genes in Foxp4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:16,674 - pyscenic.transform - WARNING - Less than 80% of the genes in Dmrta2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:17,150 - pyscenic.transform - WARNING - Less than 80% of the genes in E2f3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[######                                  ] | 16% Completed | 35min 48.1s\n",
      "2022-05-17 16:02:17,980 - pyscenic.transform - WARNING - Less than 80% of the genes in Klf11 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed | 35min 50.1s\n",
      "2022-05-17 16:02:20,002 - pyscenic.transform - WARNING - Less than 80% of the genes in Snai1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed | 35min 50.5s\n",
      "2022-05-17 16:02:20,393 - pyscenic.transform - WARNING - Less than 80% of the genes in Snapc5 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 19% Completed | 35min 51.4s\n",
      "2022-05-17 16:02:21,229 - pyscenic.transform - WARNING - Less than 80% of the genes in Sox12 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 19% Completed | 35min 51.6s\n",
      "2022-05-17 16:02:21,453 - pyscenic.transform - WARNING - Less than 80% of the genes in Sox30 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 19% Completed | 35min 51.7s\n",
      "2022-05-17 16:02:21,549 - pyscenic.transform - WARNING - Less than 80% of the genes in Pbx1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 19% Completed | 35min 54.7s\n",
      "2022-05-17 16:02:24,557 - pyscenic.transform - WARNING - Less than 80% of the genes in Pbx3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 19% Completed | 35min 55.2s\n",
      "2022-05-17 16:02:25,180 - pyscenic.transform - WARNING - Less than 80% of the genes in Pdlim5 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed | 35min 56.9s\n",
      "2022-05-17 16:02:26,739 - pyscenic.transform - WARNING - Less than 80% of the genes in Foxf1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed | 35min 59.4s\n",
      "2022-05-17 16:02:29,337 - pyscenic.transform - WARNING - Less than 80% of the genes in Zbtb24 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed | 36min  3.0s\n",
      "2022-05-17 16:02:32,931 - pyscenic.transform - WARNING - Less than 80% of the genes in Rfx8 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed | 36min  3.1s\n",
      "2022-05-17 16:02:32,986 - pyscenic.transform - WARNING - Less than 80% of the genes in Lhx2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed | 36min  3.2s\n",
      "2022-05-17 16:02:33,156 - pyscenic.transform - WARNING - Less than 80% of the genes in Rfxap could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed | 36min  4.2s\n",
      "2022-05-17 16:02:34,083 - pyscenic.transform - WARNING - Less than 80% of the genes in Sp3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed | 36min  5.8s\n",
      "2022-05-17 16:02:35,664 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp964 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed | 36min  5.9s\n",
      "2022-05-17 16:02:35,801 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp965 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed | 36min  6.6s\n",
      "2022-05-17 16:02:36,524 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp970 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:36,575 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm14327 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed | 36min  7.1s\n",
      "2022-05-17 16:02:36,995 - pyscenic.transform - WARNING - Less than 80% of the genes in Nuak1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed | 36min  7.8s\n",
      "2022-05-17 16:02:37,663 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm14443 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:37,700 - pyscenic.transform - WARNING - Less than 80% of the genes in Zbtb38 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed | 36min  8.0s\n",
      "2022-05-17 16:02:37,887 - pyscenic.transform - WARNING - Less than 80% of the genes in Scmh1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 21% Completed | 36min  8.3s\n",
      "2022-05-17 16:02:38,182 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm2000 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 22% Completed | 36min  8.7s\n",
      "2022-05-17 16:02:38,567 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm28047 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 22% Completed | 36min  8.9s\n",
      "2022-05-17 16:02:38,781 - pyscenic.transform - WARNING - Less than 80% of the genes in Zkscan3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 22% Completed | 36min  9.0s\n",
      "2022-05-17 16:02:38,863 - pyscenic.transform - WARNING - Less than 80% of the genes in Pbx1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:38,892 - pyscenic.transform - WARNING - Less than 80% of the genes in Zkscan4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[########                                ] | 22% Completed | 36min  9.3s\n",
      "2022-05-17 16:02:39,198 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm38394 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 22% Completed | 36min 10.1s\n",
      "2022-05-17 16:02:39,977 - pyscenic.transform - WARNING - Less than 80% of the genes in Pdlim5 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 22% Completed | 36min 10.6s\n",
      "2022-05-17 16:02:40,475 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp213 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 22% Completed | 36min 10.7s\n",
      "2022-05-17 16:02:40,538 - pyscenic.transform - WARNING - Less than 80% of the genes in Phf2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 22% Completed | 36min 11.1s\n",
      "2022-05-17 16:02:41,001 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp235 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 22% Completed | 36min 17.1s\n",
      "2022-05-17 16:02:47,081 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm9833 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 17.9s\n",
      "2022-05-17 16:02:47,786 - pyscenic.transform - WARNING - Less than 80% of the genes in Zscan12 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 18.6s\n",
      "2022-05-17 16:02:48,425 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp280d could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 19.0s\n",
      "2022-05-17 16:02:48,872 - pyscenic.transform - WARNING - Less than 80% of the genes in Zscan29 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 19.1s\n",
      "2022-05-17 16:02:49,034 - pyscenic.transform - WARNING - Less than 80% of the genes in Vps4b could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 19.3s\n",
      "2022-05-17 16:02:49,201 - pyscenic.transform - WARNING - Less than 80% of the genes in Edn1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 19.5s\n",
      "2022-05-17 16:02:49,343 - pyscenic.transform - WARNING - Less than 80% of the genes in Zxdc could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 20.1s\n",
      "2022-05-17 16:02:50,024 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm14443 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 20.2s\n",
      "2022-05-17 16:02:50,130 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp319 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 21.0s\n",
      "2022-05-17 16:02:50,858 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp622 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 21.3s\n",
      "2022-05-17 16:02:51,599 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp369 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 23.3s\n",
      "2022-05-17 16:02:53,216 - pyscenic.transform - WARNING - Less than 80% of the genes in Smap2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 24.1s\n",
      "2022-05-17 16:02:53,971 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp672 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 25.3s\n",
      "2022-05-17 16:02:55,155 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp445 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 25.5s\n",
      "2022-05-17 16:02:55,394 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp704 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed | 36min 26.1s\n",
      "2022-05-17 16:02:55,986 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp710 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed | 36min 27.0s\n",
      "2022-05-17 16:02:56,893 - pyscenic.transform - WARNING - Less than 80% of the genes in Pou2af1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed | 36min 28.7s\n",
      "2022-05-17 16:02:58,632 - pyscenic.transform - WARNING - Less than 80% of the genes in H1fx could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 24% Completed | 36min 29.5s\n",
      "2022-05-17 16:02:59,324 - pyscenic.transform - WARNING - Less than 80% of the genes in 9130023H24Rik could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:02:59,362 - pyscenic.transform - WARNING - Less than 80% of the genes in Egr3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed | 36min 32.6s\n",
      "2022-05-17 16:03:02,502 - pyscenic.transform - WARNING - Less than 80% of the genes in Foxp4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed | 36min 32.7s\n",
      "2022-05-17 16:03:02,644 - pyscenic.transform - WARNING - Less than 80% of the genes in Foxr1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed | 36min 33.9s\n",
      "2022-05-17 16:03:03,794 - pyscenic.transform - WARNING - Less than 80% of the genes in Cd59b could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed | 36min 34.7s\n",
      "2022-05-17 16:03:04,598 - pyscenic.transform - WARNING - Less than 80% of the genes in Ppp1r10 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed | 36min 35.7s\n",
      "2022-05-17 16:03:05,711 - pyscenic.transform - WARNING - Less than 80% of the genes in Prdm11 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed | 36min 36.8s\n",
      "2022-05-17 16:03:06,697 - pyscenic.transform - WARNING - Less than 80% of the genes in Arg2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed | 36min 37.1s\n",
      "2022-05-17 16:03:06,971 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm14443 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[##########                              ] | 25% Completed | 36min 37.6s\n",
      "2022-05-17 16:03:07,525 - pyscenic.transform - WARNING - Less than 80% of the genes in Hoxb2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed | 36min 38.0s\n",
      "2022-05-17 16:03:07,918 - pyscenic.transform - WARNING - Less than 80% of the genes in Lef1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 25% Completed | 36min 38.5s\n",
      "2022-05-17 16:03:08,398 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm28047 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed | 36min 39.4s\n",
      "2022-05-17 16:03:09,304 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp74 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed | 36min 42.1s\n",
      "2022-05-17 16:03:11,908 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp787 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed | 36min 42.2s\n",
      "2022-05-17 16:03:12,045 - pyscenic.transform - WARNING - Less than 80% of the genes in Gabpb1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed | 36min 42.3s\n",
      "2022-05-17 16:03:12,196 - pyscenic.transform - WARNING - Less than 80% of the genes in Gadd45a could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed | 36min 44.7s\n",
      "2022-05-17 16:03:14,606 - pyscenic.transform - WARNING - Less than 80% of the genes in Irx3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###########                             ] | 29% Completed | 36min 49.1s\n",
      "2022-05-17 16:03:18,980 - pyscenic.transform - WARNING - Less than 80% of the genes in Msra could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###########                             ] | 29% Completed | 36min 51.3s\n",
      "2022-05-17 16:03:21,249 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm10093 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed | 36min 51.8s\n",
      "2022-05-17 16:03:21,717 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm12166 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed | 36min 52.0s\n",
      "2022-05-17 16:03:21,884 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm12184 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed | 36min 52.3s\n",
      "2022-05-17 16:03:22,178 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp941 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed | 36min 54.4s\n",
      "2022-05-17 16:03:24,413 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm14443 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed | 36min 55.6s\n",
      "2022-05-17 16:03:25,519 - pyscenic.transform - WARNING - Less than 80% of the genes in Map4k2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed | 37min  1.1s\n",
      "2022-05-17 16:03:31,095 - pyscenic.transform - WARNING - Less than 80% of the genes in Cebpg could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed | 37min  2.8s\n",
      "2022-05-17 16:03:32,715 - pyscenic.transform - WARNING - Less than 80% of the genes in Cfl2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed | 37min  6.0s\n",
      "2022-05-17 16:03:35,898 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm9833 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed | 37min  6.6s\n",
      "2022-05-17 16:03:36,585 - pyscenic.transform - WARNING - Less than 80% of the genes in Mbnl2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 32% Completed | 37min  7.8s\n",
      "2022-05-17 16:03:37,618 - pyscenic.transform - WARNING - Less than 80% of the genes in Nuak1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#############                           ] | 32% Completed | 37min 12.1s\n",
      "2022-05-17 16:03:42,067 - pyscenic.transform - WARNING - Less than 80% of the genes in Mybl1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed | 37min 12.7s\n",
      "2022-05-17 16:03:42,587 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp74 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed | 37min 13.9s\n",
      "2022-05-17 16:03:43,805 - pyscenic.transform - WARNING - Less than 80% of the genes in Stau2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed | 37min 14.8s\n",
      "2022-05-17 16:03:44,734 - pyscenic.transform - WARNING - Less than 80% of the genes in Rfx8 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#############                           ] | 34% Completed | 37min 16.1s\n",
      "2022-05-17 16:03:45,936 - pyscenic.transform - WARNING - Less than 80% of the genes in Creb1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##############                          ] | 35% Completed | 37min 17.2s\n",
      "2022-05-17 16:03:47,037 - pyscenic.transform - WARNING - Less than 80% of the genes in Edn1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##############                          ] | 35% Completed | 37min 22.6s\n",
      "2022-05-17 16:03:52,494 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp941 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed | 37min 24.8s\n",
      "2022-05-17 16:03:54,678 - pyscenic.transform - WARNING - Less than 80% of the genes in Stat5b could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed | 37min 25.6s\n",
      "2022-05-17 16:03:55,514 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp970 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##############                          ] | 37% Completed | 37min 28.3s\n",
      "2022-05-17 16:03:58,252 - pyscenic.transform - WARNING - Less than 80% of the genes in Hes5 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed | 37min 32.3s\n",
      "2022-05-17 16:04:02,170 - pyscenic.transform - WARNING - Less than 80% of the genes in Taf7 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###############                         ] | 39% Completed | 37min 38.2s\n",
      "2022-05-17 16:04:08,096 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp558 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[################                        ] | 41% Completed | 37min 47.5s\n",
      "2022-05-17 16:04:17,421 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp558 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################                        ] | 41% Completed | 37min 49.9s\n",
      "2022-05-17 16:04:19,880 - pyscenic.transform - WARNING - Less than 80% of the genes in Cd59b could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[################                        ] | 41% Completed | 37min 50.2s\n",
      "2022-05-17 16:04:20,118 - pyscenic.transform - WARNING - Less than 80% of the genes in Prdm16 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[################                        ] | 42% Completed | 37min 51.0s\n",
      "2022-05-17 16:04:20,926 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Pou2af1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################                        ] | 42% Completed | 37min 51.5s\n",
      "2022-05-17 16:04:21,347 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Snai1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################                        ] | 42% Completed | 37min 53.6s\n",
      "2022-05-17 16:04:23,474 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zkscan4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################                        ] | 42% Completed | 37min 57.5s\n",
      "2022-05-17 16:04:27,416 - pyscenic.transform - WARNING - Less than 80% of the genes in Tceal5 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#################                       ] | 42% Completed | 38min  8.3s\n",
      "2022-05-17 16:04:38,158 - pyscenic.transform - WARNING - Less than 80% of the genes in Myog could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#################                       ] | 44% Completed | 38min 13.1s\n",
      "2022-05-17 16:04:42,956 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Foxf1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#################                       ] | 44% Completed | 38min 13.5s\n",
      "2022-05-17 16:04:43,361 - pyscenic.transform - WARNING - Less than 80% of the genes in Tceal5 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#################                       ] | 44% Completed | 38min 13.8s\n",
      "2022-05-17 16:04:43,688 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Irx3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##################                      ] | 45% Completed | 38min 23.0s\n",
      "2022-05-17 16:04:52,921 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Dmrta2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##################                      ] | 45% Completed | 38min 24.3s\n",
      "2022-05-17 16:04:54,178 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp964 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##################                      ] | 45% Completed | 38min 25.0s\n",
      "2022-05-17 16:04:54,877 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp970 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##################                      ] | 45% Completed | 38min 27.2s\n",
      "2022-05-17 16:04:57,112 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zkscan4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##################                      ] | 45% Completed | 38min 27.6s\n",
      "2022-05-17 16:04:57,520 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp369 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##################                      ] | 45% Completed | 38min 34.3s\n",
      "2022-05-17 16:05:04,203 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Dmrta2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##################                      ] | 45% Completed | 38min 35.8s\n",
      "2022-05-17 16:05:05,683 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Dmrta2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##################                      ] | 45% Completed | 38min 39.0s\n",
      "2022-05-17 16:05:08,825 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp433 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################                     ] | 48% Completed | 38min 49.9s\n",
      "2022-05-17 16:05:19,819 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp551 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################                     ] | 49% Completed | 38min 57.4s\n",
      "2022-05-17 16:05:27,320 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Foxf1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################                     ] | 49% Completed | 39min  4.3s\n",
      "2022-05-17 16:05:34,212 - pyscenic.transform - WARNING - Less than 80% of the genes in Thap11 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####################                    ] | 51% Completed | 39min 13.8s\n",
      "2022-05-17 16:05:43,692 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm10093 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 52% Completed | 39min 20.7s\n",
      "2022-05-17 16:05:50,632 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Cd59b could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 52% Completed | 39min 22.3s\n",
      "2022-05-17 16:05:52,230 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zkscan4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 52% Completed | 39min 26.2s\n",
      "2022-05-17 16:05:56,111 - pyscenic.transform - WARNING - Less than 80% of the genes in 2610044O15Rik8 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 52% Completed | 39min 26.7s\n",
      "2022-05-17 16:05:56,638 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Nuak1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 52% Completed | 39min 27.3s\n",
      "2022-05-17 16:05:57,165 - pyscenic.transform - WARNING - Less than 80% of the genes in 9130023H24Rik could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 52% Completed | 39min 28.8s\n",
      "2022-05-17 16:05:58,743 - pyscenic.transform - WARNING - Less than 80% of the genes in Agap2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 52% Completed | 39min 30.0s\n",
      "2022-05-17 16:05:59,890 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm14443 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 53% Completed | 39min 31.9s\n",
      "2022-05-17 16:06:01,748 - pyscenic.transform - WARNING - Less than 80% of the genes in Ascc1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 53% Completed | 39min 34.5s\n",
      "2022-05-17 16:06:04,383 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp551 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 53% Completed | 39min 36.8s\n",
      "2022-05-17 16:06:06,774 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Edn1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 53% Completed | 39min 39.5s\n",
      "2022-05-17 16:06:09,431 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm3055 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 54% Completed | 39min 40.3s\n",
      "2022-05-17 16:06:10,124 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm3854 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:06:10,184 - pyscenic.transform - WARNING - Less than 80% of the genes in Dbx2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 54% Completed | 39min 41.0s\n",
      "2022-05-17 16:06:10,988 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm9833 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 54% Completed | 39min 41.3s\n",
      "2022-05-17 16:06:11,213 - pyscenic.transform - WARNING - Less than 80% of the genes in E430018J23Rik could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 54% Completed | 39min 43.2s\n",
      "2022-05-17 16:06:13,106 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Edn1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 55% Completed | 39min 44.2s\n",
      "2022-05-17 16:06:14,053 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm17655 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:06:14,122 - pyscenic.transform - WARNING - Less than 80% of the genes in Ebf1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 55% Completed | 39min 44.4s\n",
      "2022-05-17 16:06:14,262 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm20939 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:06:14,267 - pyscenic.transform - WARNING - Less than 80% of the genes in Ebf4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 55% Completed | 39min 46.5s\n",
      "2022-05-17 16:06:16,405 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm14443 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 55% Completed | 39min 47.3s\n",
      "2022-05-17 16:06:17,234 - pyscenic.transform - WARNING - Less than 80% of the genes in Fosl1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed | 39min 50.2s\n",
      "2022-05-17 16:06:20,103 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Edn1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:06:20,122 - pyscenic.transform - WARNING - Less than 80% of the genes in Mnt could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed | 39min 51.2s\n",
      "2022-05-17 16:06:21,058 - pyscenic.transform - WARNING - Less than 80% of the genes in Msi2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed | 39min 51.3s\n",
      "2022-05-17 16:06:21,212 - pyscenic.transform - WARNING - Less than 80% of the genes in Msrb3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed | 39min 51.6s\n",
      "2022-05-17 16:06:21,553 - pyscenic.transform - WARNING - Less than 80% of the genes in Mterf1a could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed | 39min 51.9s\n",
      "2022-05-17 16:06:21,839 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm3055 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed | 39min 52.3s\n",
      "2022-05-17 16:06:22,191 - pyscenic.transform - WARNING - Less than 80% of the genes in Mxd1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed | 39min 52.4s\n",
      "2022-05-17 16:06:22,361 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm38394 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed | 39min 53.2s\n",
      "2022-05-17 16:06:23,123 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm6710 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed | 39min 53.3s\n",
      "2022-05-17 16:06:23,256 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm9833 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 57% Completed | 39min 55.5s\n",
      "2022-05-17 16:06:25,369 - pyscenic.transform - WARNING - Less than 80% of the genes in Phtf1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 57% Completed | 39min 55.6s\n",
      "2022-05-17 16:06:25,518 - pyscenic.transform - WARNING - Less than 80% of the genes in Pick1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 57% Completed | 39min 56.0s\n",
      "2022-05-17 16:06:25,915 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm3854 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 57% Completed | 39min 56.8s\n",
      "2022-05-17 16:06:26,656 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm9833 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 57% Completed | 39min 58.2s\n",
      "2022-05-17 16:06:28,136 - pyscenic.transform - WARNING - Less than 80% of the genes in Atf4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 57% Completed | 39min 58.6s\n",
      "2022-05-17 16:06:28,472 - pyscenic.transform - WARNING - Less than 80% of the genes in Atf6 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 57% Completed | 39min 58.7s\n",
      "2022-05-17 16:06:28,630 - pyscenic.transform - WARNING - Less than 80% of the genes in Atf6b could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 57% Completed | 39min 59.1s\n",
      "2022-05-17 16:06:28,977 - pyscenic.transform - WARNING - Less than 80% of the genes in Atoh8 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 57% Completed | 39min 59.5s\n",
      "2022-05-17 16:06:29,403 - pyscenic.transform - WARNING - Less than 80% of the genes in BC005561 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[#######################                 ] | 57% Completed | 40min  1.8s\n",
      "2022-05-17 16:06:31,718 - pyscenic.transform - WARNING - Less than 80% of the genes in Cd59b could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 57% Completed | 40min  4.1s\n",
      "2022-05-17 16:06:33,989 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp964 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 57% Completed | 40min  4.9s\n",
      "2022-05-17 16:06:34,780 - pyscenic.transform - WARNING - Less than 80% of the genes in Pou2af1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 58% Completed | 40min  8.0s\n",
      "2022-05-17 16:06:37,958 - pyscenic.transform - WARNING - Less than 80% of the genes in Bax could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 58% Completed | 40min  8.7s\n",
      "2022-05-17 16:06:38,525 - pyscenic.transform - WARNING - Less than 80% of the genes in Irf3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 58% Completed | 40min  9.3s\n",
      "2022-05-17 16:06:39,155 - pyscenic.transform - WARNING - Less than 80% of the genes in Myog could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 59% Completed | 40min 13.2s\n",
      "2022-05-17 16:06:43,068 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Cfl2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 59% Completed | 40min 14.2s\n",
      "2022-05-17 16:06:44,117 - pyscenic.transform - WARNING - Less than 80% of the genes in Zbtb42 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 59% Completed | 40min 14.3s\n",
      "2022-05-17 16:06:44,167 - pyscenic.transform - WARNING - Less than 80% of the genes in Nfatc1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 59% Completed | 40min 15.2s\n",
      "2022-05-17 16:06:45,056 - pyscenic.transform - WARNING - Less than 80% of the genes in Zbtb49 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 59% Completed | 40min 15.3s\n",
      "2022-05-17 16:06:45,179 - pyscenic.transform - WARNING - Less than 80% of the genes in Zbtb5 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 60% Completed | 40min 17.9s\n",
      "2022-05-17 16:06:47,793 - pyscenic.transform - WARNING - Less than 80% of the genes in Tcf7l2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 60% Completed | 40min 22.0s\n",
      "2022-05-17 16:06:51,882 - pyscenic.transform - WARNING - Less than 80% of the genes in Ppp1r10 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 60% Completed | 40min 22.8s\n",
      "2022-05-17 16:06:52,759 - pyscenic.transform - WARNING - Less than 80% of the genes in Prdm11 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 60% Completed | 40min 23.3s\n",
      "2022-05-17 16:06:53,232 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp433 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 24.9s\n",
      "2022-05-17 16:06:54,841 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp105 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 26.1s\n",
      "2022-05-17 16:06:56,052 - pyscenic.transform - WARNING - Less than 80% of the genes in Tet1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 26.3s\n",
      "2022-05-17 16:06:56,219 - pyscenic.transform - WARNING - Less than 80% of the genes in Bhlha15 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 26.7s\n",
      "2022-05-17 16:06:56,614 - pyscenic.transform - WARNING - Less than 80% of the genes in Gtpbp1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 26.9s\n",
      "2022-05-17 16:06:56,855 - pyscenic.transform - WARNING - Less than 80% of the genes in Gzf1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 27.0s\n",
      "2022-05-17 16:06:56,966 - pyscenic.transform - WARNING - Less than 80% of the genes in H1fx could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 27.6s\n",
      "2022-05-17 16:06:57,549 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp90 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 27.9s\n",
      "2022-05-17 16:06:57,871 - pyscenic.transform - WARNING - Less than 80% of the genes in Cfl2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 28.4s\n",
      "2022-05-17 16:06:58,290 - pyscenic.transform - WARNING - Less than 80% of the genes in Arg2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 29.1s\n",
      "2022-05-17 16:06:59,025 - pyscenic.transform - WARNING - Less than 80% of the genes in Tfap2a could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 30.3s\n",
      "2022-05-17 16:07:00,233 - pyscenic.transform - WARNING - Less than 80% of the genes in Prdm4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 32.1s\n",
      "2022-05-17 16:07:01,920 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp941 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 33.2s\n",
      "2022-05-17 16:07:03,062 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm14443 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 34.0s\n",
      "2022-05-17 16:07:03,919 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp114 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 34.1s\n",
      "2022-05-17 16:07:04,022 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp551 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 40min 35.0s\n",
      "2022-05-17 16:07:04,825 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp131 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 62% Completed | 40min 36.1s\n",
      "2022-05-17 16:07:06,037 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp592 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 62% Completed | 40min 37.5s\n",
      "2022-05-17 16:07:07,343 - pyscenic.transform - WARNING - Less than 80% of the genes in Tfeb could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 62% Completed | 40min 37.8s\n",
      "2022-05-17 16:07:07,686 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp143 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########################               ] | 62% Completed | 40min 37.9s\n",
      "2022-05-17 16:07:07,860 - pyscenic.transform - WARNING - Less than 80% of the genes in Tgif2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########################               ] | 62% Completed | 40min 38.1s\n",
      "2022-05-17 16:07:08,034 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp148 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########################               ] | 62% Completed | 40min 38.4s\n",
      "2022-05-17 16:07:08,221 - pyscenic.transform - WARNING - Less than 80% of the genes in Irx3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########################               ] | 63% Completed | 40min 40.8s\n",
      "2022-05-17 16:07:10,731 - pyscenic.transform - WARNING - Less than 80% of the genes in Thap11 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########################               ] | 63% Completed | 40min 41.6s\n",
      "2022-05-17 16:07:11,538 - pyscenic.transform - WARNING - Less than 80% of the genes in Jdp2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########################               ] | 64% Completed | 40min 45.2s\n",
      "2022-05-17 16:07:15,145 - pyscenic.transform - WARNING - Less than 80% of the genes in Hdx could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########################               ] | 64% Completed | 40min 45.3s\n",
      "2022-05-17 16:07:15,241 - pyscenic.transform - WARNING - Less than 80% of the genes in Creb1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########################               ] | 64% Completed | 40min 45.5s\n",
      "2022-05-17 16:07:15,473 - pyscenic.transform - WARNING - Less than 80% of the genes in Scmh1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########################               ] | 64% Completed | 40min 45.9s\n",
      "2022-05-17 16:07:15,818 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########################               ] | 64% Completed | 40min 47.0s\n",
      "2022-05-17 16:07:16,840 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp235 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:07:16,882 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp704 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########################               ] | 64% Completed | 40min 47.8s\n",
      "2022-05-17 16:07:17,823 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp729b could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 48.5s\n",
      "2022-05-17 16:07:18,344 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp964 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 49.1s\n",
      "2022-05-17 16:07:19,038 - pyscenic.transform - WARNING - Less than 80% of the genes in Tmem33 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 49.8s\n",
      "2022-05-17 16:07:19,641 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp296 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 50.0s\n",
      "2022-05-17 16:07:19,922 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfx could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 50.1s\n",
      "2022-05-17 16:07:19,997 - pyscenic.transform - WARNING - Less than 80% of the genes in Trim24 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 50.6s\n",
      "2022-05-17 16:07:20,480 - pyscenic.transform - WARNING - Less than 80% of the genes in Foxf1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 50.7s\n",
      "2022-05-17 16:07:20,594 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp319 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 51.3s\n",
      "2022-05-17 16:07:21,123 - pyscenic.transform - WARNING - Less than 80% of the genes in Zkscan3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 51.4s\n",
      "2022-05-17 16:07:21,274 - pyscenic.transform - WARNING - Less than 80% of the genes in Zkscan4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 51.6s\n",
      "2022-05-17 16:07:21,433 - pyscenic.transform - WARNING - Less than 80% of the genes in Zkscan5 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 52.3s\n",
      "2022-05-17 16:07:22,193 - pyscenic.transform - WARNING - Less than 80% of the genes in Hdx could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 52.5s\n",
      "2022-05-17 16:07:22,326 - pyscenic.transform - WARNING - Less than 80% of the genes in Zscan12 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 52.7s\n",
      "2022-05-17 16:07:22,593 - pyscenic.transform - WARNING - Less than 80% of the genes in Hes5 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 54.2s\n",
      "2022-05-17 16:07:24,130 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp362 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 54.4s\n",
      "2022-05-17 16:07:24,266 - pyscenic.transform - WARNING - Less than 80% of the genes in Hic2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 54.5s\n",
      "2022-05-17 16:07:24,459 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp369 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 40min 54.6s\n",
      "2022-05-17 16:07:24,541 - pyscenic.transform - WARNING - Less than 80% of the genes in Hic2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:07:24,550 - pyscenic.transform - WARNING - Less than 80% of the genes in Esrra could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:07:24,593 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp382 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[##########################              ] | 66% Completed | 40min 55.7s\n",
      "2022-05-17 16:07:25,621 - pyscenic.transform - WARNING - Less than 80% of the genes in Nf1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 66% Completed | 40min 56.0s\n",
      "2022-05-17 16:07:25,959 - pyscenic.transform - WARNING - Less than 80% of the genes in Nfatc1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 66% Completed | 40min 56.2s\n",
      "2022-05-17 16:07:26,045 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp746 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:07:26,046 - pyscenic.transform - WARNING - Less than 80% of the genes in Atf6 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 66% Completed | 40min 59.4s\n",
      "2022-05-17 16:07:29,237 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp775 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 66% Completed | 40min 59.9s\n",
      "2022-05-17 16:07:29,855 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp787 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 66% Completed | 41min  1.8s\n",
      "2022-05-17 16:07:31,670 - pyscenic.transform - WARNING - Less than 80% of the genes in Zscan29 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 66% Completed | 41min  1.9s\n",
      "2022-05-17 16:07:31,793 - pyscenic.transform - WARNING - Less than 80% of the genes in Atoh8 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 66% Completed | 41min  2.2s\n",
      "2022-05-17 16:07:32,131 - pyscenic.transform - WARNING - Less than 80% of the genes in Zxdc could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 67% Completed | 41min  2.5s\n",
      "2022-05-17 16:07:32,452 - pyscenic.transform - WARNING - Less than 80% of the genes in Smad2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 67% Completed | 41min  3.1s\n",
      "2022-05-17 16:07:32,968 - pyscenic.transform - WARNING - Less than 80% of the genes in Snai1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 67% Completed | 41min  3.4s\n",
      "2022-05-17 16:07:33,336 - pyscenic.transform - WARNING - Less than 80% of the genes in Smap2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 67% Completed | 41min  3.8s\n",
      "2022-05-17 16:07:33,732 - pyscenic.transform - WARNING - Less than 80% of the genes in Hivep3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed | 41min  4.8s\n",
      "2022-05-17 16:07:34,616 - pyscenic.transform - WARNING - Less than 80% of the genes in Snai1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed | 41min  5.0s\n",
      "2022-05-17 16:07:34,908 - pyscenic.transform - WARNING - Less than 80% of the genes in 2610044O15Rik8 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed | 41min  5.6s\n",
      "2022-05-17 16:07:35,585 - pyscenic.transform - WARNING - Less than 80% of the genes in Dbx2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed | 41min  6.2s\n",
      "2022-05-17 16:07:36,114 - pyscenic.transform - WARNING - Less than 80% of the genes in 9130023H24Rik could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min  6.8s\n",
      "2022-05-17 16:07:36,635 - pyscenic.transform - WARNING - Less than 80% of the genes in AW146154 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min  7.4s\n",
      "2022-05-17 16:07:37,281 - pyscenic.transform - WARNING - Less than 80% of the genes in Aco1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min  7.6s\n",
      "2022-05-17 16:07:37,564 - pyscenic.transform - WARNING - Less than 80% of the genes in Nfyc could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min  8.1s\n",
      "2022-05-17 16:07:37,992 - pyscenic.transform - WARNING - Less than 80% of the genes in Dmrta2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min  8.3s\n",
      "2022-05-17 16:07:38,200 - pyscenic.transform - WARNING - Less than 80% of the genes in Nnt could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min  9.8s\n",
      "2022-05-17 16:07:39,656 - pyscenic.transform - WARNING - Less than 80% of the genes in Dus3l could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min 10.1s\n",
      "2022-05-17 16:07:39,944 - pyscenic.transform - WARNING - Less than 80% of the genes in Vps4b could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min 10.2s\n",
      "2022-05-17 16:07:40,102 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp213 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min 10.3s\n",
      "2022-05-17 16:07:40,203 - pyscenic.transform - WARNING - Less than 80% of the genes in Wdr83 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min 10.4s\n",
      "2022-05-17 16:07:40,348 - pyscenic.transform - WARNING - Less than 80% of the genes in Wrnip1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min 11.7s\n",
      "2022-05-17 16:07:41,604 - pyscenic.transform - WARNING - Less than 80% of the genes in Snapc5 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min 12.3s\n",
      "2022-05-17 16:07:42,166 - pyscenic.transform - WARNING - Less than 80% of the genes in Nr2c2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min 12.7s\n",
      "2022-05-17 16:07:42,665 - pyscenic.transform - WARNING - Less than 80% of the genes in Sox12 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min 12.8s\n",
      "2022-05-17 16:07:42,752 - pyscenic.transform - WARNING - Less than 80% of the genes in Snapc4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min 13.6s\n",
      "2022-05-17 16:07:43,536 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Tal1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 41min 13.8s\n",
      "2022-05-17 16:07:43,739 - pyscenic.transform - WARNING - Less than 80% of the genes in Sox12 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[############################            ] | 70% Completed | 41min 14.9s\n",
      "2022-05-17 16:07:44,867 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp964 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed | 41min 15.2s\n",
      "2022-05-17 16:07:45,007 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp965 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed | 41min 15.8s\n",
      "2022-05-17 16:07:45,752 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp970 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed | 41min 15.9s\n",
      "2022-05-17 16:07:45,855 - pyscenic.transform - WARNING - Less than 80% of the genes in Sox30 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed | 41min 17.2s\n",
      "2022-05-17 16:07:47,112 - pyscenic.transform - WARNING - Less than 80% of the genes in Hoxa2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:07:47,139 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp235 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed | 41min 17.3s\n",
      "2022-05-17 16:07:47,285 - pyscenic.transform - WARNING - Less than 80% of the genes in Hoxa3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 18.1s\n",
      "2022-05-17 16:07:48,039 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp622 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:07:48,060 - pyscenic.transform - WARNING - Less than 80% of the genes in Zkscan3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 18.3s\n",
      "2022-05-17 16:07:48,226 - pyscenic.transform - WARNING - Less than 80% of the genes in Zkscan4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 19.7s\n",
      "2022-05-17 16:07:49,644 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp280d could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 19.9s\n",
      "2022-05-17 16:07:49,728 - pyscenic.transform - WARNING - Less than 80% of the genes in Zscan12 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 20.6s\n",
      "2022-05-17 16:07:50,497 - pyscenic.transform - WARNING - Less than 80% of the genes in Hoxa6 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 21.2s\n",
      "2022-05-17 16:07:51,115 - pyscenic.transform - WARNING - Less than 80% of the genes in Zscan29 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 21.4s\n",
      "2022-05-17 16:07:51,285 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp672 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 21.6s\n",
      "2022-05-17 16:07:51,563 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp319 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 21.7s\n",
      "2022-05-17 16:07:51,625 - pyscenic.transform - WARNING - Less than 80% of the genes in Zxdc could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 22.2s\n",
      "2022-05-17 16:07:52,081 - pyscenic.transform - WARNING - Less than 80% of the genes in Hoxa6 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 22.4s\n",
      "2022-05-17 16:07:52,349 - pyscenic.transform - WARNING - Less than 80% of the genes in Klf11 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 22.8s\n",
      "2022-05-17 16:07:52,684 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp704 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 23.4s\n",
      "2022-05-17 16:07:53,283 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp710 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 24.0s\n",
      "2022-05-17 16:07:53,956 - pyscenic.transform - WARNING - Less than 80% of the genes in 9130023H24Rik could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 25.1s\n",
      "2022-05-17 16:07:55,076 - pyscenic.transform - WARNING - Less than 80% of the genes in Hoxa7 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 25.5s\n",
      "2022-05-17 16:07:55,347 - pyscenic.transform - WARNING - Less than 80% of the genes in Pbx1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 28.4s\n",
      "2022-05-17 16:07:58,247 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Snai1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:07:58,313 - pyscenic.transform - WARNING - Less than 80% of the genes in Rb1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 29.3s\n",
      "2022-05-17 16:07:59,197 - pyscenic.transform - WARNING - Less than 80% of the genes in Foxp4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 29.6s\n",
      "2022-05-17 16:07:59,551 - pyscenic.transform - WARNING - Less than 80% of the genes in Lhx2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 31.1s\n",
      "2022-05-17 16:08:00,996 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp369 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 32.2s\n",
      "2022-05-17 16:08:02,152 - pyscenic.transform - WARNING - Less than 80% of the genes in Vps4b could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 33.0s\n",
      "2022-05-17 16:08:02,936 - pyscenic.transform - WARNING - Less than 80% of the genes in E2f3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 33.2s\n",
      "2022-05-17 16:08:03,156 - pyscenic.transform - WARNING - Less than 80% of the genes in Arg2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 33.3s\n",
      "2022-05-17 16:08:03,189 - pyscenic.transform - WARNING - Less than 80% of the genes in Pdlim5 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 33.4s\n",
      "2022-05-17 16:08:03,334 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp74 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 41min 36.0s\n",
      "2022-05-17 16:08:05,947 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp787 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#############################           ] | 72% Completed | 41min 42.5s\n",
      "2022-05-17 16:08:12,453 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp445 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#############################           ] | 72% Completed | 41min 43.7s\n",
      "2022-05-17 16:08:13,652 - pyscenic.transform - WARNING - Less than 80% of the genes in Cxxc1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed | 41min 48.0s\n",
      "2022-05-17 16:08:17,879 - pyscenic.transform - WARNING - Less than 80% of the genes in Zbtb24 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed | 41min 48.8s\n",
      "2022-05-17 16:08:18,692 - pyscenic.transform - WARNING - Less than 80% of the genes in Nuak1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:08:18,710 - pyscenic.transform - WARNING - Less than 80% of the genes in Edn1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed | 41min 49.1s\n",
      "2022-05-17 16:08:18,998 - pyscenic.transform - WARNING - Less than 80% of the genes in Zbtb38 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed | 41min 51.3s\n",
      "2022-05-17 16:08:21,159 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp941 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed | 41min 53.4s\n",
      "2022-05-17 16:08:23,308 - pyscenic.transform - WARNING - Less than 80% of the genes in Pbx1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed | 41min 54.0s\n",
      "2022-05-17 16:08:23,917 - pyscenic.transform - WARNING - Less than 80% of the genes in Dmrta2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed | 41min 54.5s\n",
      "2022-05-17 16:08:24,431 - pyscenic.transform - WARNING - Less than 80% of the genes in Pdlim5 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed | 41min 55.1s\n",
      "2022-05-17 16:08:25,039 - pyscenic.transform - WARNING - Less than 80% of the genes in Phf2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed | 41min 55.2s\n",
      "2022-05-17 16:08:25,052 - pyscenic.transform - WARNING - Less than 80% of the genes in Rfx8 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 75% Completed | 41min 55.4s\n",
      "2022-05-17 16:08:25,331 - pyscenic.transform - WARNING - Less than 80% of the genes in Rfxap could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed | 41min 58.8s\n",
      "2022-05-17 16:08:28,657 - pyscenic.transform - WARNING - Less than 80% of the genes in Sp3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed | 41min 59.0s\n",
      "2022-05-17 16:08:28,843 - pyscenic.transform - WARNING - Less than 80% of the genes in Lef1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed | 41min 59.2s\n",
      "2022-05-17 16:08:29,048 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm14327 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed | 42min  1.0s\n",
      "2022-05-17 16:08:30,974 - pyscenic.transform - WARNING - Less than 80% of the genes in Egr3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 77% Completed | 42min  6.3s\n",
      "2022-05-17 16:08:36,117 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm14443 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed | 42min  6.8s\n",
      "2022-05-17 16:08:36,724 - pyscenic.transform - WARNING - Less than 80% of the genes in Hoxb2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 77% Completed | 42min  7.3s\n",
      "2022-05-17 16:08:37,185 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm14443 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed | 42min  7.8s\n",
      "2022-05-17 16:08:37,706 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm2000 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed | 42min 13.2s\n",
      "2022-05-17 16:08:43,112 - pyscenic.transform - WARNING - Less than 80% of the genes in Pou2af1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed | 42min 14.3s\n",
      "2022-05-17 16:08:44,240 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp74 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed | 42min 15.7s\n",
      "2022-05-17 16:08:45,623 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm38394 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################################        ] | 80% Completed | 42min 16.1s\n",
      "2022-05-17 16:08:46,050 - pyscenic.transform - WARNING - Less than 80% of the genes in Irx3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################################        ] | 80% Completed | 42min 16.6s\n",
      "2022-05-17 16:08:46,547 - pyscenic.transform - WARNING - Less than 80% of the genes in Map4k2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################################        ] | 80% Completed | 42min 21.3s\n",
      "2022-05-17 16:08:51,268 - pyscenic.transform - WARNING - Less than 80% of the genes in Ppp1r10 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################################        ] | 80% Completed | 42min 22.1s\n",
      "2022-05-17 16:08:52,019 - pyscenic.transform - WARNING - Less than 80% of the genes in Prdm11 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed | 42min 22.8s\n",
      "2022-05-17 16:08:52,753 - pyscenic.transform - WARNING - Less than 80% of the genes in Foxf1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed | 42min 24.4s\n",
      "2022-05-17 16:08:54,387 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm9833 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed | 42min 24.6s\n",
      "2022-05-17 16:08:54,549 - pyscenic.transform - WARNING - Less than 80% of the genes in Cd59b could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[################################        ] | 81% Completed | 42min 25.2s\n",
      "2022-05-17 16:08:55,123 - pyscenic.transform - WARNING - Less than 80% of the genes in Nuak1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################################        ] | 81% Completed | 42min 28.2s\n",
      "2022-05-17 16:08:58,044 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp558 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed | 42min 31.8s\n",
      "2022-05-17 16:09:01,715 - pyscenic.transform - WARNING - Less than 80% of the genes in Mbnl2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed | 42min 34.5s\n",
      "2022-05-17 16:09:04,369 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Foxf1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed | 42min 35.2s\n",
      "2022-05-17 16:09:05,080 - pyscenic.transform - WARNING - Less than 80% of the genes in Msra could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################################        ] | 82% Completed | 42min 38.1s\n",
      "2022-05-17 16:09:08,022 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp941 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed | 42min 40.5s\n",
      "2022-05-17 16:09:10,408 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp970 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed | 42min 40.7s\n",
      "2022-05-17 16:09:10,619 - pyscenic.transform - WARNING - Less than 80% of the genes in Rfx8 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed | 42min 43.5s\n",
      "2022-05-17 16:09:13,401 - pyscenic.transform - WARNING - Less than 80% of the genes in Edn1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#################################       ] | 84% Completed | 42min 44.9s\n",
      "2022-05-17 16:09:14,890 - pyscenic.transform - WARNING - Less than 80% of the genes in Mybl1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#################################       ] | 84% Completed | 42min 50.7s\n",
      "2022-05-17 16:09:20,534 - pyscenic.transform - WARNING - Less than 80% of the genes in Cebpg could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#################################       ] | 84% Completed | 42min 52.2s\n",
      "2022-05-17 16:09:22,067 - pyscenic.transform - WARNING - Less than 80% of the genes in Cfl2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##################################      ] | 87% Completed | 42min 59.4s\n",
      "2022-05-17 16:09:29,333 - pyscenic.transform - WARNING - Less than 80% of the genes in Creb1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 43min  2.4s\n",
      "2022-05-17 16:09:32,312 - pyscenic.transform - WARNING - Less than 80% of the genes in Cd59b could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 43min  3.0s\n",
      "2022-05-17 16:09:32,896 - pyscenic.transform - WARNING - Less than 80% of the genes in Foxp4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 43min  3.1s\n",
      "2022-05-17 16:09:33,005 - pyscenic.transform - WARNING - Less than 80% of the genes in Foxr1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 43min  5.3s\n",
      "2022-05-17 16:09:35,261 - pyscenic.transform - WARNING - Less than 80% of the genes in Gabpb1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 43min  5.4s\n",
      "2022-05-17 16:09:35,360 - pyscenic.transform - WARNING - Less than 80% of the genes in Scmh1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 16:09:35,368 - pyscenic.transform - WARNING - Less than 80% of the genes in Gadd45a could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 43min  7.8s\n",
      "2022-05-17 16:09:37,728 - pyscenic.transform - WARNING - Less than 80% of the genes in Myog could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 43min 10.1s\n",
      "2022-05-17 16:09:39,965 - pyscenic.transform - WARNING - Less than 80% of the genes in Hes5 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 43min 10.5s\n",
      "2022-05-17 16:09:40,330 - pyscenic.transform - WARNING - Less than 80% of the genes in Stau2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################################     ] | 89% Completed | 43min 13.6s\n",
      "2022-05-17 16:09:43,514 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm10093 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################################     ] | 89% Completed | 43min 14.0s\n",
      "2022-05-17 16:09:43,830 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm12166 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###################################     ] | 89% Completed | 43min 14.1s\n",
      "2022-05-17 16:09:43,937 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm12184 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 43min 17.7s\n",
      "2022-05-17 16:09:47,630 - pyscenic.transform - WARNING - Less than 80% of the genes in Prdm16 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 43min 18.1s\n",
      "2022-05-17 16:09:48,049 - pyscenic.transform - WARNING - Less than 80% of the genes in Taf7 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[####################################    ] | 90% Completed | 43min 23.6s\n",
      "2022-05-17 16:09:53,550 - pyscenic.transform - WARNING - Less than 80% of the genes in Smap2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[####################################    ] | 91% Completed | 43min 31.2s\n",
      "2022-05-17 16:10:01,094 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm14443 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################################   ] | 93% Completed | 43min 39.8s\n",
      "2022-05-17 16:10:09,711 - pyscenic.transform - WARNING - Less than 80% of the genes in Tceal5 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####################################   ] | 93% Completed | 43min 41.9s\n",
      "2022-05-17 16:10:11,818 - pyscenic.transform - WARNING - Less than 80% of the genes in Tceal5 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################################  ] | 95% Completed | 44min  0.9s\n",
      "2022-05-17 16:10:30,747 - pyscenic.transform - WARNING - Less than 80% of the genes in Dbx2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######################################  ] | 97% Completed | 44min 28.7s\n",
      "2022-05-17 16:10:58,607 - pyscenic.transform - WARNING - Less than 80% of the genes in Thap11 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[####################################### ] | 98% Completed | 44min 54.9s\n",
      "2022-05-17 16:11:24,849 - pyscenic.transform - WARNING - Less than 80% of the genes in Fosl1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########################################] | 100% Completed | 45min 53.4s\n",
      "\n",
      "2022-05-17 16:12:23,967 - pyscenic.cli.pyscenic - INFO - Writing results to file.\n"
     ]
    }
   ],
   "source": [
    "! docker run -it --rm \\\n",
    "    -v /home/csj/mywork/ilc/220517-pyscenic-grn:/scenicdata \\\n",
    "    aertslab/pyscenic:0.11.2 pyscenic ctx \\\n",
    "        /scenicdata/expr_mat.adjacencies_abm.tsv \\\n",
    "        /scenicdata/mm10__refseq-r80__10kb_up_and_down_tss.mc9nr.feather        \\\n",
    "        /scenicdata/mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr.feather \\\n",
    "        --annotations_fname    /scenicdata/motifs-v9-nr.mgi-m0.001-o0.0.tbl \\\n",
    "        --expression_mtx_fname /scenicdata/expr_mat_abm.tsv  \\\n",
    "        --mode \"dask_multiprocessing\" \\\n",
    "        --output /scenicdata/abm_regulons_seed1.csv \\\n",
    "        --num_workers 32\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1e1868d",
   "metadata": {},
   "source": [
    "# AUCell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "9d006ce4",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T16:21:50.295564Z",
     "start_time": "2022-05-17T16:20:15.346259Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-05-17 16:20:27,658 - pyscenic.cli.pyscenic - INFO - Loading expression matrix.\n",
      "\n",
      "2022-05-17 16:20:53,341 - pyscenic.cli.pyscenic - INFO - Loading gene signatures.\n",
      "Create regulons from a dataframe of enriched features.\n",
      "Additional columns saved: []\n",
      "\n",
      "2022-05-17 16:21:32,698 - pyscenic.cli.pyscenic - INFO - Calculating cellular enrichment.\n",
      "\n",
      "2022-05-17 16:21:47,050 - pyscenic.cli.pyscenic - INFO - Writing results to file.\n"
     ]
    }
   ],
   "source": [
    "!docker run -it --rm \\\n",
    "    -v /home/csj/mywork/ilc/220517-pyscenic-grn:/scenicdata \\\n",
    "    aertslab/pyscenic:0.11.2 pyscenic aucell \\\n",
    "        /scenicdata/expr_mat_abm.tsv \\\n",
    "        /scenicdata/abm_regulons_seed1.csv \\\n",
    "        -o /scenicdata/abm_auc_mtx.csv \\\n",
    "        --num_workers 6\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "dac733bd",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-22T16:32:27.179420Z",
     "start_time": "2022-11-22T16:32:25.809089Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "regulons_abm = pd.read_csv('./abm_regulons_seed1.csv',index_col=[0,1], header=[0,1])\n",
    "auc_abm = pd.read_csv('./abm_auc_mtx.csv')\n",
    "auc_abm= auc_abm.set_index(\"Cell\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "476ac038",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T16:43:31.827002Z",
     "start_time": "2022-05-17T16:42:12.936199Z"
    }
   },
   "outputs": [],
   "source": [
    "from pyscenic.binarization import binarize\n",
    "bin_abm, auc_thresholds = binarize( auc_abm, num_workers=25 )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "ca4af07d",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T16:44:34.668389Z",
     "start_time": "2022-05-17T16:44:34.611601Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>2010315B03Rik(+)</th>\n",
       "      <th>2610044O15Rik8(+)</th>\n",
       "      <th>Aggf1(+)</th>\n",
       "      <th>Ahr(+)</th>\n",
       "      <th>Ar(+)</th>\n",
       "      <th>Arid3a(+)</th>\n",
       "      <th>Arid5b(+)</th>\n",
       "      <th>Arnt(+)</th>\n",
       "      <th>Arnt2(+)</th>\n",
       "      <th>Arntl(+)</th>\n",
       "      <th>...</th>\n",
       "      <th>Zfp937(+)</th>\n",
       "      <th>Zfp941(+)</th>\n",
       "      <th>Zfp950(+)</th>\n",
       "      <th>Zfp959(+)</th>\n",
       "      <th>Zfp960(+)</th>\n",
       "      <th>Zfp964(+)</th>\n",
       "      <th>Zfp970(+)</th>\n",
       "      <th>Zmiz1(+)</th>\n",
       "      <th>Zscan21(+)</th>\n",
       "      <th>Zxdc(+)</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cell</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AAACCTGAGACCTTTG</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACCTGAGATGTAAC</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACCTGGTTAGGGTG</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACGGGAGCCTATGT</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACGGGAGTGTCCAT</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGGTTTCATGCATG</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGGTTTCGCGCCAA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTCACACATGACT</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTCACACCAACCG</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTCATCACTCCTG</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3442 rows Ã— 448 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  2010315B03Rik(+)  2610044O15Rik8(+)  Aggf1(+)  Ahr(+)  \\\n",
       "Cell                                                                      \n",
       "AAACCTGAGACCTTTG                 0                  0         1       0   \n",
       "AAACCTGAGATGTAAC                 0                  0         1       0   \n",
       "AAACCTGGTTAGGGTG                 0                  0         0       0   \n",
       "AAACGGGAGCCTATGT                 0                  0         0       0   \n",
       "AAACGGGAGTGTCCAT                 0                  0         0       0   \n",
       "...                            ...                ...       ...     ...   \n",
       "TTTGGTTTCATGCATG                 1                  0         0       1   \n",
       "TTTGGTTTCGCGCCAA                 0                  0         1       0   \n",
       "TTTGTCACACATGACT                 0                  1         0       0   \n",
       "TTTGTCACACCAACCG                 0                  0         0       0   \n",
       "TTTGTCATCACTCCTG                 0                  0         0       0   \n",
       "\n",
       "                  Ar(+)  Arid3a(+)  Arid5b(+)  Arnt(+)  Arnt2(+)  Arntl(+)  \\\n",
       "Cell                                                                         \n",
       "AAACCTGAGACCTTTG      0          0          0        1         1         0   \n",
       "AAACCTGAGATGTAAC      0          0          0        1         0         0   \n",
       "AAACCTGGTTAGGGTG      0          1          1        1         0         0   \n",
       "AAACGGGAGCCTATGT      0          0          0        1         1         0   \n",
       "AAACGGGAGTGTCCAT      0          0          0        1         0         0   \n",
       "...                 ...        ...        ...      ...       ...       ...   \n",
       "TTTGGTTTCATGCATG      0          0          0        1         1         0   \n",
       "TTTGGTTTCGCGCCAA      0          0          0        0         0         0   \n",
       "TTTGTCACACATGACT      0          0          0        1         0         0   \n",
       "TTTGTCACACCAACCG      0          1          0        1         0         0   \n",
       "TTTGTCATCACTCCTG      0          0          0        1         1         0   \n",
       "\n",
       "                  ...  Zfp937(+)  Zfp941(+)  Zfp950(+)  Zfp959(+)  Zfp960(+)  \\\n",
       "Cell              ...                                                          \n",
       "AAACCTGAGACCTTTG  ...          0          0          0          0          0   \n",
       "AAACCTGAGATGTAAC  ...          0          0          0          0          0   \n",
       "AAACCTGGTTAGGGTG  ...          0          0          0          1          0   \n",
       "AAACGGGAGCCTATGT  ...          0          0          0          0          0   \n",
       "AAACGGGAGTGTCCAT  ...          0          0          0          0          0   \n",
       "...               ...        ...        ...        ...        ...        ...   \n",
       "TTTGGTTTCATGCATG  ...          0          0          0          0          0   \n",
       "TTTGGTTTCGCGCCAA  ...          0          0          0          0          0   \n",
       "TTTGTCACACATGACT  ...          0          0          1          0          0   \n",
       "TTTGTCACACCAACCG  ...          0          0          0          0          0   \n",
       "TTTGTCATCACTCCTG  ...          0          0          0          0          0   \n",
       "\n",
       "                  Zfp964(+)  Zfp970(+)  Zmiz1(+)  Zscan21(+)  Zxdc(+)  \n",
       "Cell                                                                   \n",
       "AAACCTGAGACCTTTG          0          0         0           0        0  \n",
       "AAACCTGAGATGTAAC          0          0         0           1        0  \n",
       "AAACCTGGTTAGGGTG          0          0         0           0        0  \n",
       "AAACGGGAGCCTATGT          0          0         0           0        0  \n",
       "AAACGGGAGTGTCCAT          0          0         0           0        0  \n",
       "...                     ...        ...       ...         ...      ...  \n",
       "TTTGGTTTCATGCATG          0          0         0           1        0  \n",
       "TTTGGTTTCGCGCCAA          0          0         0           1        0  \n",
       "TTTGTCACACATGACT          0          1         0           1        0  \n",
       "TTTGTCACACCAACCG          0          1         0           0        0  \n",
       "TTTGTCATCACTCCTG          0          0         0           0        0  \n",
       "\n",
       "[3442 rows x 448 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bin_abm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "812e0362",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T16:46:13.575236Z",
     "start_time": "2022-05-17T16:46:10.656926Z"
    }
   },
   "outputs": [],
   "source": [
    "auc_abm.to_csv('auc_abm.csv')\n",
    "bin_abm.to_csv('bin_abm.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "da48001c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-23T08:23:25.986066Z",
     "start_time": "2022-11-23T08:23:25.981644Z"
    }
   },
   "source": [
    "# Check regulon members"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e4d68d6c",
   "metadata": {},
   "source": [
    "## Rorc (+)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "33d613a7",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-22T16:52:46.242914Z",
     "start_time": "2022-11-22T16:52:46.222404Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th colspan=\"8\" halign=\"left\">Enrichment</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>NES</th>\n",
       "      <th>MotifSimilarityQvalue</th>\n",
       "      <th>OrthologousIdentity</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Context</th>\n",
       "      <th>TargetGenes</th>\n",
       "      <th>RankAtMax</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TF</th>\n",
       "      <th>MotifID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"6\" valign=\"top\">Rorc</th>\n",
       "      <th>hocomoco__NR1D2_MOUSE.H11MO.0.A</th>\n",
       "      <td>0.148034</td>\n",
       "      <td>3.912438</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>motif similar to hocomoco__RORG_MOUSE.H11MO.0....</td>\n",
       "      <td>frozenset({'activating', 'mm10__refseq-r80__50...</td>\n",
       "      <td>[('Rorc', 1.0), ('S100a4', 12.110177336686409)...</td>\n",
       "      <td>970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__RORG_HUMAN.H11MO.0.C</th>\n",
       "      <td>0.068219</td>\n",
       "      <td>3.904211</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.891473</td>\n",
       "      <td>gene is orthologous to ENSG00000143365 in H. s...</td>\n",
       "      <td>frozenset({'weight&gt;90.0%', 'mm10__refseq-r80__...</td>\n",
       "      <td>[('Arntl', 1.3855036415735251), ('Lmo4', 10.03...</td>\n",
       "      <td>1084</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__NR1D1_HUMAN.H11MO.1.D</th>\n",
       "      <td>0.135246</td>\n",
       "      <td>3.521089</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.891473</td>\n",
       "      <td>gene is orthologous to ENSG00000143365 in H. s...</td>\n",
       "      <td>frozenset({'activating', 'top50', 'mm10__refse...</td>\n",
       "      <td>[('S100a4', 12.110177336686409), ('Lmo4', 10.0...</td>\n",
       "      <td>257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M07955</th>\n",
       "      <td>0.053981</td>\n",
       "      <td>3.394355</td>\n",
       "      <td>0.000559</td>\n",
       "      <td>0.259690</td>\n",
       "      <td>gene is orthologous to FBgn0000448 in D. melan...</td>\n",
       "      <td>frozenset({'activating', 'top50perTarget', 'mm...</td>\n",
       "      <td>[('Gm15417', 0.2573392809956352), ('Arntl', 1....</td>\n",
       "      <td>1064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>stark__AWNTGGGTCA</th>\n",
       "      <td>0.079842</td>\n",
       "      <td>3.381341</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.259690</td>\n",
       "      <td>gene is orthologous to FBgn0000448 in D. melan...</td>\n",
       "      <td>frozenset({'top5perTarget', 'activating', 'mm1...</td>\n",
       "      <td>[('Cap2', 0.7518122887405049), ('Bcl11b', 12.2...</td>\n",
       "      <td>856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale__RORA_DBD_WAWNTRGGTYAGTAGGTCA_repr</th>\n",
       "      <td>0.058301</td>\n",
       "      <td>3.098151</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.891473</td>\n",
       "      <td>gene is orthologous to ENSG00000143365 in H. s...</td>\n",
       "      <td>frozenset({'weight&gt;90.0%', 'mm10__refseq-r80__...</td>\n",
       "      <td>[('Rora', 5.387434331285859), ('Pofut1', 2.928...</td>\n",
       "      <td>1274</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Enrichment            \\\n",
       "                                                       AUC       NES   \n",
       "TF   MotifID                                                           \n",
       "Rorc hocomoco__NR1D2_MOUSE.H11MO.0.A              0.148034  3.912438   \n",
       "     hocomoco__RORG_HUMAN.H11MO.0.C               0.068219  3.904211   \n",
       "     hocomoco__NR1D1_HUMAN.H11MO.1.D              0.135246  3.521089   \n",
       "     transfac_pro__M07955                         0.053981  3.394355   \n",
       "     stark__AWNTGGGTCA                            0.079842  3.381341   \n",
       "     taipale__RORA_DBD_WAWNTRGGTYAGTAGGTCA_repr   0.058301  3.098151   \n",
       "\n",
       "                                                                       \\\n",
       "                                                MotifSimilarityQvalue   \n",
       "TF   MotifID                                                            \n",
       "Rorc hocomoco__NR1D2_MOUSE.H11MO.0.A                         0.000032   \n",
       "     hocomoco__RORG_HUMAN.H11MO.0.C                          0.000000   \n",
       "     hocomoco__NR1D1_HUMAN.H11MO.1.D                         0.000017   \n",
       "     transfac_pro__M07955                                    0.000559   \n",
       "     stark__AWNTGGGTCA                                       0.000000   \n",
       "     taipale__RORA_DBD_WAWNTRGGTYAGTAGGTCA_repr              0.000000   \n",
       "\n",
       "                                                                     \\\n",
       "                                                OrthologousIdentity   \n",
       "TF   MotifID                                                          \n",
       "Rorc hocomoco__NR1D2_MOUSE.H11MO.0.A                       1.000000   \n",
       "     hocomoco__RORG_HUMAN.H11MO.0.C                        0.891473   \n",
       "     hocomoco__NR1D1_HUMAN.H11MO.1.D                       0.891473   \n",
       "     transfac_pro__M07955                                  0.259690   \n",
       "     stark__AWNTGGGTCA                                     0.259690   \n",
       "     taipale__RORA_DBD_WAWNTRGGTYAGTAGGTCA_repr            0.891473   \n",
       "\n",
       "                                                                                                    \\\n",
       "                                                                                        Annotation   \n",
       "TF   MotifID                                                                                         \n",
       "Rorc hocomoco__NR1D2_MOUSE.H11MO.0.A             motif similar to hocomoco__RORG_MOUSE.H11MO.0....   \n",
       "     hocomoco__RORG_HUMAN.H11MO.0.C              gene is orthologous to ENSG00000143365 in H. s...   \n",
       "     hocomoco__NR1D1_HUMAN.H11MO.1.D             gene is orthologous to ENSG00000143365 in H. s...   \n",
       "     transfac_pro__M07955                        gene is orthologous to FBgn0000448 in D. melan...   \n",
       "     stark__AWNTGGGTCA                           gene is orthologous to FBgn0000448 in D. melan...   \n",
       "     taipale__RORA_DBD_WAWNTRGGTYAGTAGGTCA_repr  gene is orthologous to ENSG00000143365 in H. s...   \n",
       "\n",
       "                                                                                                    \\\n",
       "                                                                                           Context   \n",
       "TF   MotifID                                                                                         \n",
       "Rorc hocomoco__NR1D2_MOUSE.H11MO.0.A             frozenset({'activating', 'mm10__refseq-r80__50...   \n",
       "     hocomoco__RORG_HUMAN.H11MO.0.C              frozenset({'weight>90.0%', 'mm10__refseq-r80__...   \n",
       "     hocomoco__NR1D1_HUMAN.H11MO.1.D             frozenset({'activating', 'top50', 'mm10__refse...   \n",
       "     transfac_pro__M07955                        frozenset({'activating', 'top50perTarget', 'mm...   \n",
       "     stark__AWNTGGGTCA                           frozenset({'top5perTarget', 'activating', 'mm1...   \n",
       "     taipale__RORA_DBD_WAWNTRGGTYAGTAGGTCA_repr  frozenset({'weight>90.0%', 'mm10__refseq-r80__...   \n",
       "\n",
       "                                                                                                    \\\n",
       "                                                                                       TargetGenes   \n",
       "TF   MotifID                                                                                         \n",
       "Rorc hocomoco__NR1D2_MOUSE.H11MO.0.A             [('Rorc', 1.0), ('S100a4', 12.110177336686409)...   \n",
       "     hocomoco__RORG_HUMAN.H11MO.0.C              [('Arntl', 1.3855036415735251), ('Lmo4', 10.03...   \n",
       "     hocomoco__NR1D1_HUMAN.H11MO.1.D             [('S100a4', 12.110177336686409), ('Lmo4', 10.0...   \n",
       "     transfac_pro__M07955                        [('Gm15417', 0.2573392809956352), ('Arntl', 1....   \n",
       "     stark__AWNTGGGTCA                           [('Cap2', 0.7518122887405049), ('Bcl11b', 12.2...   \n",
       "     taipale__RORA_DBD_WAWNTRGGTYAGTAGGTCA_repr  [('Rora', 5.387434331285859), ('Pofut1', 2.928...   \n",
       "\n",
       "                                                           \n",
       "                                                RankAtMax  \n",
       "TF   MotifID                                               \n",
       "Rorc hocomoco__NR1D2_MOUSE.H11MO.0.A                  970  \n",
       "     hocomoco__RORG_HUMAN.H11MO.0.C                  1084  \n",
       "     hocomoco__NR1D1_HUMAN.H11MO.1.D                  257  \n",
       "     transfac_pro__M07955                            1064  \n",
       "     stark__AWNTGGGTCA                                856  \n",
       "     taipale__RORA_DBD_WAWNTRGGTYAGTAGGTCA_repr      1274  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "selected_motifs = ['Rorc']\n",
    "df_motifs_sel = regulons_abm.iloc[ [ True if x in selected_motifs else False for x in regulons_abm.index.get_level_values('TF') ] ,:]\n",
    "df_motifs_sel = df_motifs_sel.sort_values([('Enrichment','NES')], ascending=False)\n",
    "df_motifs_sel\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "9ef803af",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-23T08:24:03.537069Z",
     "start_time": "2022-11-23T08:24:03.449154Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th colspan=\"9\" halign=\"left\">Enrichment</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>NES</th>\n",
       "      <th>MotifSimilarityQvalue</th>\n",
       "      <th>OrthologousIdentity</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Context</th>\n",
       "      <th>TargetGenes</th>\n",
       "      <th>RankAtMax</th>\n",
       "      <th>MotifLogo</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TF</th>\n",
       "      <th>MotifID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">Rorc</th>\n",
       "      <th>hocomoco__NR1D2_MOUSE.H11MO.0.A</th>\n",
       "      <td>0.148034</td>\n",
       "      <td>3.912438</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>motif similar to hocomoco__RORG_MOUSE.H11MO.0.B ('RORG_MOUSE'; q-value = 3.18e-05) which is directly annotated</td>\n",
       "      <td>frozenset({'activating', 'mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr', 'top50'})</td>\n",
       "      <td>[('Rorc', 1.0), ('S100a4', 12.110177336686409), ('Rora', 5.387434331285859), ('Fli1', 7.961083373262843), ('Lmo4', 10.03236991345168), ('Bcl11b', 12.21512031428983), ('Lingo4', 7.502422617597499), ('Lpxn', 5.660283994149342), ('Ppt2', 7.695347083910587), ('Rxrg', 8.949470364453978), ('Ahcyl2', 5.26216921997957)]</td>\n",
       "      <td>970</td>\n",
       "      <td><img src=\"http://motifcollections.aertslab.org/v9/logos/hocomoco__NR1D2_MOUSE.H11MO.0.A.png\" style=\"max-height:124px;\"></img></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__RORG_HUMAN.H11MO.0.C</th>\n",
       "      <td>0.068219</td>\n",
       "      <td>3.904211</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.891473</td>\n",
       "      <td>gene is orthologous to ENSG00000143365 in H. sapiens (identity = 89%) which is directly annotated for motif</td>\n",
       "      <td>frozenset({'weight>90.0%', 'mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr', 'activating'})</td>\n",
       "      <td>[('Arntl', 1.3855036415735251), ('Lmo4', 10.03236991345168), ('S100a4', 12.110177336686409), ('Nr1d1', 1.4873916133890428), ('Ppt2', 7.695347083910587), ('Rorc', 1.0), ('Gtpbp2', 3.424747556568786), ('Syne1', 3.188375000602932), ('Sertad2', 1.0805397347797263), ('Cnn2', 2.9444737261288125), ('Il17re', 3.41398011853336), ('Dnajb2', 0.9896701526538444), ('Espn', 2.702646285776281), ('Fam124b', 1.764016895663912), ('Lat', 2.4907086257592166), ('Nebl', 3.22307360155461), ('Tmem51', 0.9250514897173496), ('Diaph1', 1.612512077869393), ('Ccr6', 0.9085999378717208), ('Chd2', 2.357024314560804), ('Utp18', 1.7785719267299518), ('Nfia', 2.4959315316168817), ('Ptp4a2', 2.2492610876139087), ('Rhog', 1.3834112770435998), ('Camk4', 4.442368642718844), ('Ahcyl2', 5.26216921997957), ('Rxrg', 8.949470364453978), ('9130401M01Rik', 0.9155227611083784), ('Rab11fip1', 0.8333692943368006), ('Kmt5b', 2.589454493355827), ('Gm5122', 1.3337520843049238), ('Lonp2', 0.9095960765113246), ('Lpxn', 5.660283994149342), ('Ccr2', 8.958332493169546), ('Selenot', 5.75773838889312)]</td>\n",
       "      <td>1084</td>\n",
       "      <td><img src=\"http://motifcollections.aertslab.org/v9/logos/hocomoco__RORG_HUMAN.H11MO.0.C.png\" style=\"max-height:124px;\"></img></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__NR1D1_HUMAN.H11MO.1.D</th>\n",
       "      <td>0.135246</td>\n",
       "      <td>3.521089</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.891473</td>\n",
       "      <td>gene is orthologous to ENSG00000143365 in H. sapiens (identity = 89%) which is annotated for similar motif cisbp__M6455 ('RORC[gene ID: \"ENSG00000143365\" species: \"Homo sapiens\" TF status: \"direct\" TF family: \"Nuclear receptor\" DBDs: \"zf-C4\"]; Rorc[gene ID: \"ENSMUSG00000028150\" species: \"Mus musculus\" TF status: \"inferred\" TF family: \"Nuclear receptor\" DBDs: \"zf-C4\"]'; q-value = 1.67e-05)</td>\n",
       "      <td>frozenset({'activating', 'top50', 'mm10__refseq-r80__10kb_up_and_down_tss.mc9nr'})</td>\n",
       "      <td>[('S100a4', 12.110177336686409), ('Lmo4', 10.03236991345168), ('Adgrg5', 8.954202986161253), ('Zfand6', 14.635616158455395), ('Bcl11b', 12.21512031428983), ('Rorc', 1.0)]</td>\n",
       "      <td>257</td>\n",
       "      <td><img src=\"http://motifcollections.aertslab.org/v9/logos/hocomoco__NR1D1_HUMAN.H11MO.1.D.png\" style=\"max-height:124px;\"></img></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transfac_pro__M07955</th>\n",
       "      <td>0.053981</td>\n",
       "      <td>3.394355</td>\n",
       "      <td>0.000559</td>\n",
       "      <td>0.259690</td>\n",
       "      <td>gene is orthologous to FBgn0000448 in D. melanogaster (identity = 25%) which is annotated for similar motif flyfactorsurvey__Hr46_SANGER_5_FBgn0000448 ('Hr46_SANGER_5_FBgn0000448'; q-value = 0.000559)</td>\n",
       "      <td>frozenset({'activating', 'top50perTarget', 'mm10__refseq-r80__10kb_up_and_down_tss.mc9nr'})</td>\n",
       "      <td>[('Gm15417', 0.2573392809956352), ('Arntl', 1.3855036415735251), ('Adam12', 1.1432877555048568), ('Npas2', 0.1197585581448402), ('Msi2', 0.9637405563786684), ('Thra', 0.4140075062771627), ('Gpr183', 1.9510515746119088), ('Rnf2', 2.034597845110726), ('Pea15a', 0.5792929879584138), ('S100a3', 0.0558711915172876), ('Clcn3', 1.7881191638309393), ('Cited2', 2.305774276154912), ('Ndst1', 0.567194343791989), ('Map3k14', 0.1121570768829758), ('Rara', 0.166094922567329), ('Cmtm6', 0.4995561423301303), ('Il23r', 1.1764181219727082e-08), ('Nr1d1', 1.4873916133890428), ('Tspan14', 0.1987083461284088), ('Smco4', 15.399200527878705), ('Mprip', 2.678689569424614), ('Zbtb20', 0.7497570559816599), ('Trps1', 0.2199427947388579), ('S100a4', 12.110177336686409), ('Uhrf2', 1.8088663206957143), ('Rorc', 1.0), ('Eif1b', 0.1870978825113096), ('Mmd', 0.6604899071047476), ('Txk', 3.126292435749126), ('Zfp827', 0.1514271543850054), ('Sesn3', 0.3957291592993856), ('Bcl11b', 12.21512031428983), ('Sh3bp5', 0.8975896495308701), ('Akirin2', 0.7941354589517543), ('Itk', 0.3460362173362782), ('Pde4b', 0.192469969657754), ('Syne1', 3.188375000602932), ('Cdc42ep3', 1.177496299953876), ('Lmo4', 10.03236991345168), ('Mgea5', 0.2117511997482772), ('Epc1', 0.2463847684257322), ('1700061F12Rik', 0.010574699361165), ('Phf21a', 0.1926032036423581), ('Vamp5', 0.3231734475775129), ('Epb41l2', 0.0242174498147958), ('Lgals8', 0.1942933179383019), ('Jak2', 0.200253158544095), ('Hormad2', 0.067831317017486), ('Cd36', 0.0230236585017368), ('Csrnp1', 0.1727339846130305), ('Pde2a', 0.1106624506994908), ('Sorbs2', 0.1714479150501825), ('Sh3kbp1', 1.373101223490784), ('Ahr', 1.3637328322287252), ('Tdrkh', 2.118325905898596), ('Jun', 1.0360257413246747), ('Ets1', 1.707627236398234), ('Cym', 0.241089153254801), ('Ppfibp1', 0.0891775192568437), ('Tgfbr3', 0.190264192350364), ('Tmem140', 1.2094240082677656), ('Il1rl1', 0.1922606583923383), ('Pik3r1', 0.2083117814168887), ('Rarg', 0.2405071381195915), ('Sp6', 0.0900873079479337), ('Shtn1', 0.2395980336504103), ('Cast', 3.495091094457527), ('Ankrd35', 0.5692740707793783), ('Kpnb1', 0.6298642708010428), ('Arhgef28', 0.2877557624081901), ('Art4', 0.3752057232360165), ('Upp2', 0.0854226349941324), ('Slc11a2', 0.7327396229171551), ('Pxdc1', 7.539235223332924), ('Cxcr6', 0.5318748648218474), ('Rpl37a', 0.2149788263010778), ('Orc4', 8.839045030775491e-05), ('Il18r1', 30.918230992838996), ('Eya2', 0.2694421701204396)]</td>\n",
       "      <td>1064</td>\n",
       "      <td><img src=\"http://motifcollections.aertslab.org/v9/logos/transfac_pro__M07955.png\" style=\"max-height:124px;\"></img></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>stark__AWNTGGGTCA</th>\n",
       "      <td>0.079842</td>\n",
       "      <td>3.381341</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.259690</td>\n",
       "      <td>gene is orthologous to FBgn0000448 in D. melanogaster (identity = 25%) which is directly annotated for motif</td>\n",
       "      <td>frozenset({'top5perTarget', 'activating', 'mm10__refseq-r80__10kb_up_and_down_tss.mc9nr'})</td>\n",
       "      <td>[('Cap2', 0.7518122887405049), ('Bcl11b', 12.21512031428983), ('Kif21b', 1.1566411744998195), ('Rims3', 1.0650112159965082), ('Adgrg5', 8.954202986161253), ('Syne1', 3.188375000602932), ('Pkn1', 0.5774108002970881), ('Diaph1', 1.612512077869393), ('Iqsec1', 1.1291006897152385), ('Nfia', 2.4959315316168817), ('Zfand6', 14.635616158455395), ('Sertad2', 1.0805397347797263), ('Smox', 7.26762504444825), ('Mgmt', 1.2295832974509893), ('Enpp4', 2.6065224388255883), ('Plekho2', 3.036834726979893)]</td>\n",
       "      <td>856</td>\n",
       "      <td><img src=\"http://motifcollections.aertslab.org/v9/logos/stark__AWNTGGGTCA.png\" style=\"max-height:124px;\"></img></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from pyscenic.utils import load_motifs\n",
    "import operator as op\n",
    "from IPython.display import HTML, display\n",
    "\n",
    "BASE_URL = \"http://motifcollections.aertslab.org/v9/logos/\"\n",
    "COLUMN_NAME_LOGO = \"MotifLogo\"\n",
    "COLUMN_NAME_MOTIF_ID = \"MotifID\"\n",
    "COLUMN_NAME_TARGETS = \"TargetGenes\"\n",
    "\n",
    "def display_logos(df: pd.DataFrame, top_target_genes: int = 3, base_url: str = BASE_URL):\n",
    "    \"\"\"\n",
    "    :param df:\n",
    "    :param base_url:\n",
    "    \"\"\"\n",
    "    # Make sure the original dataframe is not altered.\n",
    "    df = df.copy()\n",
    "    \n",
    "    # Add column with URLs to sequence logo.\n",
    "    def create_url(motif_id):\n",
    "        return '<img src=\"{}{}.png\" style=\"max-height:124px;\"></img>'.format(base_url, motif_id)\n",
    "    df[(\"Enrichment\", COLUMN_NAME_LOGO)] = list(map(create_url, df.index.get_level_values(COLUMN_NAME_MOTIF_ID)))\n",
    "    \n",
    "    # Truncate TargetGenes.\n",
    "    def truncate(col_val):\n",
    "        return sorted(col_val, key=op.itemgetter(1))[:top_target_genes]\n",
    "    \n",
    "    #df[(\"Enrichment\", COLUMN_NAME_TARGETS)] = list(map(truncate, df[(\"Enrichment\", COLUMN_NAME_TARGETS)]))\n",
    "    \n",
    "    MAX_COL_WIDTH = pd.get_option('display.max_colwidth')\n",
    "    pd.set_option('display.max_colwidth', 200)\n",
    "    display(HTML(df.head().to_html(escape=False)))\n",
    "    pd.set_option('display.max_colwidth', MAX_COL_WIDTH)\n",
    "    \n",
    "display_logos(df_motifs_sel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "a9d46ed5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-23T08:30:06.710049Z",
     "start_time": "2022-11-23T08:30:06.689146Z"
    }
   },
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'dict_keys' object has no attribute 'tolist'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Input \u001b[0;32mIn [93]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mres\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkeys\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mtolist\u001b[49m()\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'dict_keys' object has no attribute 'tolist'"
     ]
    }
   ],
   "source": [
    "res.keys().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "f5c4a9cc",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-23T08:30:56.945887Z",
     "start_time": "2022-11-23T08:30:56.930974Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>target</th>\n",
       "      <th>importance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Rora</td>\n",
       "      <td>5.387434</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Pofut1</td>\n",
       "      <td>2.928190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Nr1d1</td>\n",
       "      <td>1.487392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fam124b</td>\n",
       "      <td>1.764017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Tecpr1</td>\n",
       "      <td>0.893833</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>Lonp2</td>\n",
       "      <td>0.909596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>Lpxn</td>\n",
       "      <td>5.660284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>Selenot</td>\n",
       "      <td>5.757738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>Fli1</td>\n",
       "      <td>7.961083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>Lingo4</td>\n",
       "      <td>7.502423</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>145 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      target  importance\n",
       "0       Rora    5.387434\n",
       "1     Pofut1    2.928190\n",
       "2      Nr1d1    1.487392\n",
       "3    Fam124b    1.764017\n",
       "4     Tecpr1    0.893833\n",
       "..       ...         ...\n",
       "140    Lonp2    0.909596\n",
       "141     Lpxn    5.660284\n",
       "142  Selenot    5.757738\n",
       "143     Fli1    7.961083\n",
       "144   Lingo4    7.502423\n",
       "\n",
       "[145 rows x 2 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "e83483e9",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-23T08:33:40.294505Z",
     "start_time": "2022-11-23T08:33:40.262201Z"
    },
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>target</th>\n",
       "      <th>importance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>Il18r1</td>\n",
       "      <td>3.091823e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>Smco4</td>\n",
       "      <td>1.539920e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Zfand6</td>\n",
       "      <td>1.463562e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Bcl11b</td>\n",
       "      <td>1.221512e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>S100a4</td>\n",
       "      <td>1.211018e+01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>Epb41l2</td>\n",
       "      <td>2.421745e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>Cd36</td>\n",
       "      <td>2.302366e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>1700061F12Rik</td>\n",
       "      <td>1.057470e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>Orc4</td>\n",
       "      <td>8.839045e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>Il23r</td>\n",
       "      <td>1.176418e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>145 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            target    importance\n",
       "121         Il18r1  3.091823e+01\n",
       "72           Smco4  1.539920e+01\n",
       "40          Zfand6  1.463562e+01\n",
       "41          Bcl11b  1.221512e+01\n",
       "37          S100a4  1.211018e+01\n",
       "..             ...           ...\n",
       "90         Epb41l2  2.421745e-02\n",
       "94            Cd36  2.302366e-02\n",
       "87   1700061F12Rik  1.057470e-02\n",
       "120           Orc4  8.839045e-05\n",
       "70           Il23r  1.176418e-08\n",
       "\n",
       "[145 rows x 2 columns]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "targets = regulons_abm.loc['Rorc',['Enrichment']].droplevel(0, axis=1)['TargetGenes']\n",
    "targets = [eval(tg) for tg in targets]\n",
    "import numpy as np\n",
    "res ={}\n",
    "for target in targets:\n",
    "    for tup in target:\n",
    "        if tup[0] in res.keys():\n",
    "            res[tup[0]].append( float(tup[1]))\n",
    "        else:\n",
    "            res[tup[0]] = [ tup[1] ]\n",
    "\n",
    "\n",
    "for k,v in res.items():\n",
    "    res[k] = np.mean(v) \n",
    "targets = pd.DataFrame({\"target\": res.keys(), \"importance\": res.values() })\\\n",
    "         .sort_values('importance',ascending=False)\n",
    "\n",
    "targets.to_csv('220517-SCENICprediction-ABM-Targets-of-Rorc.csv')\n",
    "targets"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
