{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "48da0f34",
   "metadata": {},
   "source": [
    "# GRN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1f77ce31",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T09:48:03.712933Z",
     "start_time": "2022-05-17T09:15:48.346994Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-05-17 09:15:58,006 - pyscenic.cli.pyscenic - INFO - Loading expression matrix.\n",
      "\n",
      "2022-05-17 09:16:24,723 - pyscenic.cli.pyscenic - INFO - Inferring regulatory networks.\n",
      "preparing dask client\n",
      "parsing input\n",
      "creating dask graph\n",
      "16 partitions\n",
      "computing dask graph\n",
      "not shutting down client, client was created externally\n",
      "finished\n",
      "distributed.worker - WARNING - Heartbeat to scheduler failed\n",
      "\n",
      "2022-05-17 09:47:55,168 - pyscenic.cli.pyscenic - INFO - Writing results to file.\n"
     ]
    }
   ],
   "source": [
    "!docker run -it --rm \\\n",
    "    -v /home/csj/mywork/ilc/220517-pyscenic-grn:/scenicdata \\\n",
    "    aertslab/pyscenic:0.11.2 pyscenic grn \\\n",
    "        --num_workers 16 \\\n",
    "        -o /scenicdata/expr_mat.adjacencies_fl.tsv \\\n",
    "        /scenicdata/expr_mat_fl.tsv \\\n",
    "        /scenicdata/mm_mgi_tfs.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6bb7fe89",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T10:00:17.546640Z",
     "start_time": "2022-05-17T10:00:17.167726Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 583348\r\n",
      "drwxrwxr-x  3 csj  csj       4096 5月  17 17:49 .\r\n",
      "drwxr-xr-x 18 csj  csj       4096 5月  17 16:59 ..\r\n",
      "-rw-rw-r--  1 csj  csj       2355 5月  17 17:15 220517-generate-input-for-pyscenic.ipynb\r\n",
      "-rw-rw-r--  1 csj  csj       1722 5月  17 17:18 abm.ipynb\r\n",
      "-rw-rw-r--  1 csj  csj  258112823 5月  17 17:14 expr_mat_abm.tsv\r\n",
      "-rw-r--r--  1 root root         0 5月  17 01:33 expr_mat.adjacencies_abm.tsv\r\n",
      "-rw-r--r--  1 root root  58172886 5月  17 17:48 expr_mat.adjacencies_fl.tsv\r\n",
      "-rw-rw-r--  1 csj  csj  281018745 5月  17 17:15 expr_mat_fl.tsv\r\n",
      "-rw-rw-r--  1 csj  csj       2096 5月  17 17:49 fl.ipynb\r\n",
      "drwxrwxr-x  2 csj  csj       4096 5月  17 17:13 .ipynb_checkpoints\r\n",
      "-rw-rw-r--  1 csj  csj      10775 5月  17 01:31 mm_mgi_tfs.txt\r\n"
     ]
    }
   ],
   "source": [
    "! ls -al"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f3dc1ce1",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T10:00:26.217598Z",
     "start_time": "2022-05-17T10:00:26.064482Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TF\ttarget\timportance\r\n",
      "Hmgb2\tNusap1\t168.10664067115485\r\n",
      "Hmgb2\tCenpa\t163.59710726165258\r\n",
      "Hmgb2\tBirc5\t163.55184857114907\r\n",
      "Hmgb2\tUbe2c\t157.556053264593\r\n",
      "Kif22\tUbe2c\t152.1008622451824\r\n",
      "Stat1\tTgtp1\t148.1063861068085\r\n",
      "Hmgb2\tKif22\t147.45174935861832\r\n",
      "Irf8\tPrtn3\t146.58362462325655\r\n",
      "Hmgb2\tCcnb2\t142.3652617196257\r\n"
     ]
    }
   ],
   "source": [
    "! head expr_mat.adjacencies_fl.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "efc559b1",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T11:08:25.409716Z",
     "start_time": "2022-05-17T11:08:23.802564Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "grn = pd.read_csv('expr_mat.adjacencies_fl.tsv',sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "57fcda53",
   "metadata": {},
   "source": [
    "# CTX"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "d68e5d60",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T11:57:14.314649Z",
     "start_time": "2022-05-17T11:34:44.352588Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-05-17 11:34:54,411 - pyscenic.cli.pyscenic - INFO - Creating modules.\n",
      "\n",
      "2022-05-17 11:34:55,410 - pyscenic.cli.pyscenic - INFO - Loading expression matrix.\n",
      "\n",
      "2022-05-17 11:35:24,102 - pyscenic.utils - INFO - Calculating Pearson correlations.\n",
      "\n",
      "2022-05-17 11:35:24,665 - pyscenic.utils - WARNING - Note on correlation calculation: the default behaviour for calculating the correlations has changed after pySCENIC verion 0.9.16. Previously, the default was to calculate the correlation between a TF and target gene using only cells with non-zero expression values (mask_dropouts=True). The current default is now to use all cells to match the behavior of the R verision of SCENIC. The original settings can be retained by setting 'rho_mask_dropouts=True' in the modules_from_adjacencies function, or '--mask_dropouts' from the CLI.\n",
      "\tDropout masking is currently set to [False].\n",
      "\n",
      "2022-05-17 11:36:19,728 - pyscenic.utils - INFO - Creating modules.\n",
      "\n",
      "2022-05-17 11:38:46,361 - pyscenic.cli.pyscenic - INFO - Loading databases.\n",
      "\n",
      "2022-05-17 11:38:46,362 - pyscenic.cli.pyscenic - INFO - Calculating regulons.\n",
      "[                                        ] | 0% Completed | 31.2s\n",
      "2022-05-17 11:39:19,771 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gtf3c2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 42.2s\n",
      "2022-05-17 11:39:30,816 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for 2810403A07Rik could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 47.4s\n",
      "2022-05-17 11:39:35,987 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Arg2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed | 55.5s\n",
      "2022-05-17 11:39:44,099 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm6710 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[                                        ] | 0% Completed |  1min 13.3s\n",
      "2022-05-17 11:40:01,897 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Ing3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 2% Completed |  1min 40.6s\n",
      "2022-05-17 11:40:29,101 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Cat could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 2% Completed |  1min 51.5s\n",
      "2022-05-17 11:40:40,041 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Mtf1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 3% Completed |  2min 11.3s\n",
      "2022-05-17 11:40:59,895 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Tcf12 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 3% Completed |  2min 18.3s\n",
      "2022-05-17 11:41:06,877 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Pax8 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#                                       ] | 4% Completed |  2min 24.0s\n",
      "2022-05-17 11:41:12,563 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zc3h7a could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###                                     ] | 8% Completed |  3min  6.4s\n",
      "2022-05-17 11:41:55,019 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp688 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[####                                    ] | 10% Completed |  3min 14.6s\n",
      "2022-05-17 11:42:03,161 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Foxp4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed |  3min 35.2s\n",
      "2022-05-17 11:42:23,719 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp688 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 13% Completed |  3min 36.8s\n",
      "2022-05-17 11:42:25,333 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Foxp4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 14% Completed |  3min 53.8s\n",
      "2022-05-17 11:42:42,414 - pyscenic.transform - WARNING - Less than 80% of the genes in 2810403A07Rik could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#####                                   ] | 14% Completed |  3min 58.1s\n",
      "2022-05-17 11:42:46,698 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zc3h7a could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######                                  ] | 15% Completed |  4min  3.3s\n",
      "2022-05-17 11:42:51,916 - pyscenic.transform - WARNING - Less than 80% of the genes in Cat could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######                                  ] | 15% Completed |  4min  6.6s\n",
      "2022-05-17 11:42:55,129 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm6710 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######                                  ] | 15% Completed |  4min 12.7s\n",
      "2022-05-17 11:43:01,248 - pyscenic.transform - WARNING - Less than 80% of the genes in Arntl could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[######                                  ] | 16% Completed |  4min 16.6s\n",
      "2022-05-17 11:43:05,182 - pyscenic.transform - WARNING - Less than 80% of the genes in Ing4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  4min 28.8s\n",
      "2022-05-17 11:43:17,381 - pyscenic.transform - WARNING - Less than 80% of the genes in Gtf2b could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  4min 34.0s\n",
      "2022-05-17 11:43:22,585 - pyscenic.transform - WARNING - Less than 80% of the genes in Irf3 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  4min 51.9s\n",
      "2022-05-17 11:43:40,471 - pyscenic.transform - WARNING - Less than 80% of the genes in Ncbp2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  4min 56.8s\n",
      "2022-05-17 11:43:45,384 - pyscenic.transform - WARNING - Less than 80% of the genes in Socs4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#######                                 ] | 18% Completed |  4min 57.1s\n",
      "2022-05-17 11:43:45,647 - pyscenic.transform - WARNING - Less than 80% of the genes in Esrra could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  5min 17.9s\n",
      "2022-05-17 11:44:06,434 - pyscenic.transform - WARNING - Less than 80% of the genes in Rbm8a2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  5min 19.8s\n",
      "2022-05-17 11:44:08,387 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[########                                ] | 20% Completed |  5min 27.8s\n",
      "2022-05-17 11:44:16,328 - pyscenic.transform - WARNING - Less than 80% of the genes in Cxxc1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[########                                ] | 22% Completed |  5min 42.1s\n",
      "2022-05-17 11:44:30,689 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp280d could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  5min 44.7s\n",
      "2022-05-17 11:44:33,324 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp958 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  5min 49.1s\n",
      "2022-05-17 11:44:37,675 - pyscenic.transform - WARNING - Less than 80% of the genes in Nmi could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  5min 49.4s\n",
      "2022-05-17 11:44:37,927 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Tcf12 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  5min 52.9s\n",
      "2022-05-17 11:44:41,501 - pyscenic.transform - WARNING - Less than 80% of the genes in Thap11 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  5min 54.1s\n",
      "2022-05-17 11:44:42,685 - pyscenic.transform - WARNING - Less than 80% of the genes in Rnf114 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  5min 56.4s\n",
      "2022-05-17 11:44:45,015 - pyscenic.transform - WARNING - Less than 80% of the genes in Dnajc21 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  5min 56.8s\n",
      "2022-05-17 11:44:45,425 - pyscenic.transform - WARNING - Less than 80% of the genes in Dnttip1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#########                               ] | 23% Completed |  6min  0.6s\n",
      "2022-05-17 11:44:49,100 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfpm1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  6min  8.0s\n",
      "2022-05-17 11:44:56,532 - pyscenic.transform - WARNING - Less than 80% of the genes in Tsnax could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  6min  9.2s\n",
      "2022-05-17 11:44:57,723 - pyscenic.transform - WARNING - Less than 80% of the genes in Usf1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  6min 11.7s\n",
      "2022-05-17 11:45:00,262 - pyscenic.transform - WARNING - Less than 80% of the genes in Zswim1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  6min 14.4s\n",
      "2022-05-17 11:45:02,941 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm6710 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##########                              ] | 26% Completed |  6min 17.5s\n",
      "2022-05-17 11:45:06,007 - pyscenic.transform - WARNING - Less than 80% of the genes in Foxp4 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########                             ] | 27% Completed |  6min 20.2s\n",
      "2022-05-17 11:45:08,759 - pyscenic.transform - WARNING - Less than 80% of the genes in Yy1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########                             ] | 27% Completed |  6min 24.8s\n",
      "2022-05-17 11:45:13,359 - pyscenic.transform - WARNING - Less than 80% of the genes in Taf1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########                             ] | 27% Completed |  6min 24.9s\n",
      "2022-05-17 11:45:13,508 - pyscenic.transform - WARNING - Less than 80% of the genes in Mef2a could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########                             ] | 27% Completed |  6min 27.1s\n",
      "2022-05-17 11:45:15,629 - pyscenic.transform - WARNING - Less than 80% of the genes in Gtf3c2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########                             ] | 27% Completed |  6min 29.8s\n",
      "2022-05-17 11:45:18,377 - pyscenic.transform - WARNING - Less than 80% of the genes in Zdhhc15 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  6min 31.4s\n",
      "2022-05-17 11:45:19,997 - pyscenic.transform - WARNING - Less than 80% of the genes in Pax8 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  6min 31.8s\n",
      "2022-05-17 11:45:20,383 - pyscenic.transform - WARNING - Less than 80% of the genes in Lcor could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###########                             ] | 28% Completed |  6min 38.5s\n",
      "2022-05-17 11:45:27,017 - pyscenic.transform - WARNING - Less than 80% of the genes in Tbpl1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  6min 46.9s\n",
      "2022-05-17 11:45:35,443 - pyscenic.transform - WARNING - Less than 80% of the genes in Gfi1 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  6min 52.2s\n",
      "2022-05-17 11:45:40,764 - pyscenic.transform - WARNING - Less than 80% of the genes in Tcf12 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 30% Completed |  6min 54.4s\n",
      "2022-05-17 11:45:42,992 - pyscenic.transform - WARNING - Less than 80% of the genes in Zdhhc15 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  7min  5.0s\n",
      "2022-05-17 11:45:53,635 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp958 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[############                            ] | 31% Completed |  7min  7.3s\n",
      "2022-05-17 11:45:55,889 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp445 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[#############                           ] | 33% Completed |  7min 19.6s\n",
      "2022-05-17 11:46:08,228 - pyscenic.transform - WARNING - Less than 80% of the genes in Mef2a could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[##############                          ] | 36% Completed |  7min 49.0s\n",
      "2022-05-17 11:46:37,542 - pyscenic.transform - WARNING - Less than 80% of the genes in Nmi could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[###############                         ] | 38% Completed |  8min 27.5s\n",
      "2022-05-17 11:47:16,039 - pyscenic.transform - WARNING - Less than 80% of the genes in Srebf2 could be mapped to mm10__refseq-r80__10kb_up_and_down_tss.mc9nr. Skipping this module.\n",
      "[################                        ] | 41% Completed |  8min 47.5s\n",
      "2022-05-17 11:47:36,087 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gtf3c2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###################                     ] | 49% Completed | 10min 17.0s\n",
      "2022-05-17 11:49:05,658 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for 2810403A07Rik could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###################                     ] | 49% Completed | 10min 17.5s\n",
      "2022-05-17 11:49:05,995 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Pax8 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####################                    ] | 50% Completed | 10min 27.0s\n",
      "2022-05-17 11:49:15,626 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Gm6710 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[####################                    ] | 51% Completed | 10min 30.9s\n",
      "2022-05-17 11:49:19,411 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Arg2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#####################                   ] | 53% Completed | 10min 38.8s\n",
      "2022-05-17 11:49:27,375 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Ing3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 55% Completed | 10min 50.5s\n",
      "2022-05-17 11:49:39,146 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zc3h7a could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 55% Completed | 10min 52.6s\n",
      "2022-05-17 11:49:41,165 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Cat could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed | 10min 58.4s\n",
      "2022-05-17 11:49:46,921 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Tcf12 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[######################                  ] | 56% Completed | 10min 58.6s\n",
      "2022-05-17 11:49:47,138 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Mtf1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#######################                 ] | 58% Completed | 11min 29.1s\n",
      "2022-05-17 11:50:17,667 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp688 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 60% Completed | 11min 34.7s\n",
      "2022-05-17 11:50:23,236 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Foxp4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########################                ] | 61% Completed | 12min 14.2s\n",
      "2022-05-17 11:51:02,798 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zc3h7a could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 12min 44.6s\n",
      "2022-05-17 11:51:33,187 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Zfp688 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 65% Completed | 12min 46.3s\n",
      "2022-05-17 11:51:34,909 - pyscenic.transform - WARNING - Less than 80% of the genes in 2810403A07Rik could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 67% Completed | 12min 53.6s\n",
      "2022-05-17 11:51:42,213 - pyscenic.transform - WARNING - Less than 80% of the genes in Cat could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 67% Completed | 12min 54.1s\n",
      "2022-05-17 11:51:42,590 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm6710 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##########################              ] | 67% Completed | 12min 55.7s\n",
      "2022-05-17 11:51:44,280 - pyscenic.transform - WARNING - Less than 80% of the genes in Ing4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed | 12min 57.9s\n",
      "2022-05-17 11:51:46,427 - pyscenic.transform - WARNING - Less than 80% of the genes in Mtf1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed | 12min 58.1s\n",
      "2022-05-17 11:51:46,717 - pyscenic.transform - WARNING - Less than 80% of the genes in Gtf2b could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed | 13min  0.9s\n",
      "2022-05-17 11:51:49,411 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Foxp4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed | 13min  5.0s\n",
      "2022-05-17 11:51:53,576 - pyscenic.transform - WARNING - Less than 80% of the genes in Irf3 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 68% Completed | 13min  9.8s\n",
      "2022-05-17 11:51:58,436 - pyscenic.transform - WARNING - Less than 80% of the genes in Arntl could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###########################             ] | 69% Completed | 13min 14.6s\n",
      "2022-05-17 11:52:03,217 - pyscenic.transform - WARNING - Less than 80% of the genes in Esrra could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 70% Completed | 13min 18.9s\n",
      "2022-05-17 11:52:07,560 - pyscenic.transform - WARNING - Less than 80% of the genes in Poli could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 13min 23.9s\n",
      "2022-05-17 11:52:12,511 - pyscenic.transform - WARNING - Less than 80% of the genes in Ncbp2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 13min 28.9s\n",
      "2022-05-17 11:52:17,441 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 13min 32.3s\n",
      "2022-05-17 11:52:20,927 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp958 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 13min 32.5s\n",
      "2022-05-17 11:52:21,041 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp959 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 13min 34.0s\n",
      "2022-05-17 11:52:22,601 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfpm1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 13min 35.3s\n",
      "2022-05-17 11:52:23,940 - pyscenic.transform - WARNING - Less than 80% of the genes in Thap11 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[############################            ] | 71% Completed | 13min 39.0s\n",
      "2022-05-17 11:52:27,599 - pyscenic.transform - WARNING - Less than 80% of the genes in Regulon for Tcf12 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#############################           ] | 72% Completed | 13min 43.5s\n",
      "2022-05-17 11:52:32,113 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp207 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 11:52:32,123 - pyscenic.transform - WARNING - Less than 80% of the genes in Zswim1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[#############################           ] | 72% Completed | 13min 44.4s\n",
      "2022-05-17 11:52:32,941 - pyscenic.transform - WARNING - Less than 80% of the genes in Tsnax could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed | 13min 46.1s\n",
      "2022-05-17 11:52:34,682 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp280d could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed | 13min 49.8s\n",
      "2022-05-17 11:52:38,343 - pyscenic.transform - WARNING - Less than 80% of the genes in Socs4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed | 13min 51.7s\n",
      "2022-05-17 11:52:40,223 - pyscenic.transform - WARNING - Less than 80% of the genes in Rbm8a2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed | 13min 53.4s\n",
      "2022-05-17 11:52:41,873 - pyscenic.transform - WARNING - Less than 80% of the genes in Usf1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#############################           ] | 73% Completed | 14min  1.3s\n",
      "2022-05-17 11:52:49,869 - pyscenic.transform - WARNING - Less than 80% of the genes in Nmi could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed | 14min  6.9s\n",
      "2022-05-17 11:52:55,476 - pyscenic.transform - WARNING - Less than 80% of the genes in Cxxc1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed | 14min 12.9s\n",
      "2022-05-17 11:53:01,417 - pyscenic.transform - WARNING - Less than 80% of the genes in Yy1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed | 14min 14.7s\n",
      "2022-05-17 11:53:03,215 - pyscenic.transform - WARNING - Less than 80% of the genes in Mef2a could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[##############################          ] | 76% Completed | 14min 16.6s\n",
      "2022-05-17 11:53:05,161 - pyscenic.transform - WARNING - Less than 80% of the genes in Foxp4 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed | 14min 21.2s\n",
      "2022-05-17 11:53:09,741 - pyscenic.transform - WARNING - Less than 80% of the genes in Dnajc21 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed | 14min 21.6s\n",
      "2022-05-17 11:53:10,180 - pyscenic.transform - WARNING - Less than 80% of the genes in Dnttip1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed | 14min 25.7s\n",
      "2022-05-17 11:53:14,223 - pyscenic.transform - WARNING - Less than 80% of the genes in Zbtb7b could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed | 14min 26.3s\n",
      "2022-05-17 11:53:14,851 - pyscenic.transform - WARNING - Less than 80% of the genes in Zdhhc15 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed | 14min 27.8s\n",
      "2022-05-17 11:53:16,380 - pyscenic.transform - WARNING - Less than 80% of the genes in Gm6710 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed | 14min 30.1s\n",
      "2022-05-17 11:53:18,623 - pyscenic.transform - WARNING - Less than 80% of the genes in Gfi1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed | 14min 36.4s\n",
      "2022-05-17 11:53:24,919 - pyscenic.transform - WARNING - Less than 80% of the genes in Gtf3c2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "\n",
      "2022-05-17 11:53:24,995 - pyscenic.transform - WARNING - Less than 80% of the genes in Pax8 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 78% Completed | 14min 37.1s\n",
      "2022-05-17 11:53:25,685 - pyscenic.transform - WARNING - Less than 80% of the genes in Rnf114 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed | 14min 45.8s\n",
      "2022-05-17 11:53:34,383 - pyscenic.transform - WARNING - Less than 80% of the genes in Lcor could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed | 14min 46.6s\n",
      "2022-05-17 11:53:35,204 - pyscenic.transform - WARNING - Less than 80% of the genes in Zdhhc15 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###############################         ] | 79% Completed | 14min 48.9s\n",
      "2022-05-17 11:53:37,496 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp445 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#################################       ] | 82% Completed | 15min  8.0s\n",
      "2022-05-17 11:53:56,589 - pyscenic.transform - WARNING - Less than 80% of the genes in Mef2a could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed | 15min 16.4s\n",
      "2022-05-17 11:54:04,972 - pyscenic.transform - WARNING - Less than 80% of the genes in Nmi could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed | 15min 17.5s\n",
      "2022-05-17 11:54:06,044 - pyscenic.transform - WARNING - Less than 80% of the genes in Taf1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#################################       ] | 83% Completed | 15min 18.8s\n",
      "2022-05-17 11:54:07,332 - pyscenic.transform - WARNING - Less than 80% of the genes in Zfp958 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#################################       ] | 84% Completed | 15min 31.1s\n",
      "2022-05-17 11:54:19,620 - pyscenic.transform - WARNING - Less than 80% of the genes in Tbpl1 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[#################################       ] | 84% Completed | 15min 33.5s\n",
      "2022-05-17 11:54:22,061 - pyscenic.transform - WARNING - Less than 80% of the genes in Tcf12 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[###################################     ] | 88% Completed | 16min  8.1s\n",
      "2022-05-17 11:54:56,749 - pyscenic.transform - WARNING - Less than 80% of the genes in Srebf2 could be mapped to mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr. Skipping this module.\n",
      "[########################################] | 100% Completed | 18min 21.7s\n",
      "\n",
      "2022-05-17 11:57:10,418 - pyscenic.cli.pyscenic - INFO - Writing results to file.\n"
     ]
    }
   ],
   "source": [
    "! docker run -it --rm \\\n",
    "    -v /home/csj/mywork/ilc/220517-pyscenic-grn:/scenicdata \\\n",
    "    aertslab/pyscenic:0.11.2 pyscenic ctx \\\n",
    "        /scenicdata/expr_mat.adjacencies_fl.tsv \\\n",
    "        /scenicdata/mm10__refseq-r80__10kb_up_and_down_tss.mc9nr.feather        \\\n",
    "        /scenicdata/mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr.feather \\\n",
    "        --annotations_fname    /scenicdata/motifs-v9-nr.mgi-m0.001-o0.0.tbl \\\n",
    "        --expression_mtx_fname /scenicdata/expr_mat_fl.tsv  \\\n",
    "        --mode \"dask_multiprocessing\" \\\n",
    "        --output /scenicdata/fl_regulons_seed1.csv \\\n",
    "        --num_workers 16\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "da6dc133",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T11:34:42.072187Z",
     "start_time": "2022-05-17T11:34:41.898367Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/csj/mywork/ilc/220517-pyscenic-grn/motifs-v9-nr.mgi-m0.001-o0.0.tbl\r\n"
     ]
    }
   ],
   "source": [
    "! ls /home/csj/mywork/ilc/220517-pyscenic-grn/motifs-v9-nr.mgi-m0.001-o0.0.tbl"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "49e144b6",
   "metadata": {},
   "source": [
    "# AUCell"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "6ccba22c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T15:59:35.855051Z",
     "start_time": "2022-05-17T15:40:59.039179Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2022-05-17 15:41:08,150 - pyscenic.cli.pyscenic - INFO - Loading expression matrix.\n",
      "\n",
      "2022-05-17 15:46:30,578 - pyscenic.cli.pyscenic - INFO - Loading gene signatures.\n",
      "Create regulons from a dataframe of enriched features.\n",
      "Additional columns saved: []\n",
      "\n",
      "2022-05-17 15:46:54,573 - pyscenic.cli.pyscenic - INFO - Calculating cellular enrichment.\n",
      "\n",
      "2022-05-17 15:59:29,801 - pyscenic.cli.pyscenic - INFO - Writing results to file.\n"
     ]
    }
   ],
   "source": [
    "!docker run -it --rm \\\n",
    "    -v /home/csj/mywork/ilc/220517-pyscenic-grn:/scenicdata \\\n",
    "    aertslab/pyscenic:0.11.2 pyscenic aucell \\\n",
    "        /scenicdata/expr_mat_fl.tsv \\\n",
    "        /scenicdata/fl_regulons_seed1.csv \\\n",
    "        -o /scenicdata/fl_auc_mtx.csv \\\n",
    "        --num_workers 6\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "5477af02",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-22T16:25:06.114523Z",
     "start_time": "2022-11-22T16:24:59.188623Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "regulons_fl = pd.read_csv('./fl_regulons_seed1.csv',index_col=[0,1], header=[0,1])\n",
    "auc_fl = pd.read_csv('./fl_auc_mtx.csv')\n",
    "auc_fl= auc_fl.set_index('Cell')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "bf84fd21",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T16:42:33.577238Z",
     "start_time": "2022-05-17T16:41:24.617080Z"
    }
   },
   "outputs": [],
   "source": [
    "from pyscenic.binarization import binarize\n",
    "bin_fl, auc_thresholds = binarize( auc_fl, num_workers=25 )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "0c7ef269",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T16:44:27.510854Z",
     "start_time": "2022-05-17T16:44:27.484924Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ar(+)</th>\n",
       "      <th>Arid3a(+)</th>\n",
       "      <th>Arnt(+)</th>\n",
       "      <th>Arnt2(+)</th>\n",
       "      <th>Atf1(+)</th>\n",
       "      <th>Atf3(+)</th>\n",
       "      <th>Atf4(+)</th>\n",
       "      <th>Atf6(+)</th>\n",
       "      <th>Atf6b(+)</th>\n",
       "      <th>Atf7(+)</th>\n",
       "      <th>...</th>\n",
       "      <th>Zfp94(+)</th>\n",
       "      <th>Zfp950(+)</th>\n",
       "      <th>Zfp97(+)</th>\n",
       "      <th>Zfp970(+)</th>\n",
       "      <th>Zfp974(+)</th>\n",
       "      <th>Zfp976(+)</th>\n",
       "      <th>Zkscan14(+)</th>\n",
       "      <th>Zmiz1(+)</th>\n",
       "      <th>Zrsr1(+)</th>\n",
       "      <th>Zscan29(+)</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cell</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AAACCTGAGAGCTGGT</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACCTGAGCCCTAAT</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACCTGAGCTGGAAC</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACCTGAGGAACTGC</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACCTGAGGAGTTTA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTCACACATGTGT</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTCACATACTCTT</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTCATCCGCAGTG</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTCATCGCCCTTA</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTCATCGTACGGC</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4284 rows × 422 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Ar(+)  Arid3a(+)  Arnt(+)  Arnt2(+)  Atf1(+)  Atf3(+)  \\\n",
       "Cell                                                                      \n",
       "AAACCTGAGAGCTGGT      0          0        0         0        0        0   \n",
       "AAACCTGAGCCCTAAT      0          0        0         0        0        0   \n",
       "AAACCTGAGCTGGAAC      0          0        0         0        1        0   \n",
       "AAACCTGAGGAACTGC      0          0        0         0        1        0   \n",
       "AAACCTGAGGAGTTTA      0          0        0         0        0        0   \n",
       "...                 ...        ...      ...       ...      ...      ...   \n",
       "TTTGTCACACATGTGT      0          0        0         0        1        0   \n",
       "TTTGTCACATACTCTT      0          0        0         0        1        0   \n",
       "TTTGTCATCCGCAGTG      0          0        0         0        1        0   \n",
       "TTTGTCATCGCCCTTA      0          0        0         0        0        0   \n",
       "TTTGTCATCGTACGGC      0          0        0         0        0        1   \n",
       "\n",
       "                  Atf4(+)  Atf6(+)  Atf6b(+)  Atf7(+)  ...  Zfp94(+)  \\\n",
       "Cell                                                   ...             \n",
       "AAACCTGAGAGCTGGT        0        0         0        0  ...         0   \n",
       "AAACCTGAGCCCTAAT        0        0         0        0  ...         1   \n",
       "AAACCTGAGCTGGAAC        1        1         0        1  ...         0   \n",
       "AAACCTGAGGAACTGC        0        1         0        0  ...         0   \n",
       "AAACCTGAGGAGTTTA        0        0         0        0  ...         1   \n",
       "...                   ...      ...       ...      ...  ...       ...   \n",
       "TTTGTCACACATGTGT        1        0         0        1  ...         0   \n",
       "TTTGTCACATACTCTT        0        0         0        0  ...         0   \n",
       "TTTGTCATCCGCAGTG        0        0         0        0  ...         1   \n",
       "TTTGTCATCGCCCTTA        1        0         0        0  ...         0   \n",
       "TTTGTCATCGTACGGC        1        0         0        0  ...         0   \n",
       "\n",
       "                  Zfp950(+)  Zfp97(+)  Zfp970(+)  Zfp974(+)  Zfp976(+)  \\\n",
       "Cell                                                                     \n",
       "AAACCTGAGAGCTGGT          0         0          0          0          0   \n",
       "AAACCTGAGCCCTAAT          1         0          0          0          0   \n",
       "AAACCTGAGCTGGAAC          0         0          0          0          0   \n",
       "AAACCTGAGGAACTGC          1         0          0          0          0   \n",
       "AAACCTGAGGAGTTTA          0         0          0          0          0   \n",
       "...                     ...       ...        ...        ...        ...   \n",
       "TTTGTCACACATGTGT          0         0          0          0          0   \n",
       "TTTGTCACATACTCTT          1         0          0          0          0   \n",
       "TTTGTCATCCGCAGTG          0         0          0          0          0   \n",
       "TTTGTCATCGCCCTTA          0         0          0          0          0   \n",
       "TTTGTCATCGTACGGC          0         0          0          0          0   \n",
       "\n",
       "                  Zkscan14(+)  Zmiz1(+)  Zrsr1(+)  Zscan29(+)  \n",
       "Cell                                                           \n",
       "AAACCTGAGAGCTGGT            0         0         0           0  \n",
       "AAACCTGAGCCCTAAT            0         0         0           0  \n",
       "AAACCTGAGCTGGAAC            0         0         0           0  \n",
       "AAACCTGAGGAACTGC            0         1         0           0  \n",
       "AAACCTGAGGAGTTTA            0         0         0           0  \n",
       "...                       ...       ...       ...         ...  \n",
       "TTTGTCACACATGTGT            0         1         0           0  \n",
       "TTTGTCACATACTCTT            0         1         0           0  \n",
       "TTTGTCATCCGCAGTG            0         0         0           0  \n",
       "TTTGTCATCGCCCTTA            0         0         0           0  \n",
       "TTTGTCATCGTACGGC            0         0         0           0  \n",
       "\n",
       "[4284 rows x 422 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bin_fl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "3a3c26b7",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-17T16:45:53.724860Z",
     "start_time": "2022-05-17T16:45:50.599254Z"
    }
   },
   "outputs": [],
   "source": [
    "auc_fl.to_csv('auc_fl.csv')\n",
    "bin_fl.to_csv('bin_fl.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77dbb5df",
   "metadata": {},
   "source": [
    "# Check regulon members"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6c452919",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-22T16:25:09.560257Z",
     "start_time": "2022-11-22T16:25:09.470516Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th colspan=\"8\" halign=\"left\">Enrichment</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>NES</th>\n",
       "      <th>MotifSimilarityQvalue</th>\n",
       "      <th>OrthologousIdentity</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Context</th>\n",
       "      <th>TargetGenes</th>\n",
       "      <th>RankAtMax</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TF</th>\n",
       "      <th>MotifID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Ar</th>\n",
       "      <th>transfac_pro__M08908</th>\n",
       "      <td>0.073686</td>\n",
       "      <td>3.037615</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.877642</td>\n",
       "      <td>gene is orthologous to ENSG00000169083 in H. s...</td>\n",
       "      <td>frozenset({'activating', 'weight&gt;75.0%', 'mm10...</td>\n",
       "      <td>[('Ar', 1.0), ('Rdh12', 0.4101501149570101), (...</td>\n",
       "      <td>1359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">Arnt</th>\n",
       "      <th>jaspar__MA0603.1</th>\n",
       "      <td>0.057200</td>\n",
       "      <td>3.073739</td>\n",
       "      <td>0.000799</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>gene is annotated for similar motif transfac_p...</td>\n",
       "      <td>frozenset({'activating', 'weight&gt;75.0%', 'mm10...</td>\n",
       "      <td>[('Rnf115', 1.1655901315529902), ('Klf6', 0.93...</td>\n",
       "      <td>471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__USF1_MOUSE.H11MO.0.A</th>\n",
       "      <td>0.066981</td>\n",
       "      <td>3.951676</td>\n",
       "      <td>0.000675</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>gene is annotated for similar motif swissregul...</td>\n",
       "      <td>frozenset({'activating', 'weight&gt;75.0%', 'mm10...</td>\n",
       "      <td>[('Rnf115', 1.1655901315529902), ('Map3k12', 0...</td>\n",
       "      <td>1118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taipale_cyt_meth__ARNTL_RTCACGTGMN_eDBD</th>\n",
       "      <td>0.057509</td>\n",
       "      <td>3.101489</td>\n",
       "      <td>0.000479</td>\n",
       "      <td>0.389381</td>\n",
       "      <td>gene is orthologous to FBgn0264075 in D. melan...</td>\n",
       "      <td>frozenset({'activating', 'weight&gt;75.0%', 'mm10...</td>\n",
       "      <td>[('Rnf115', 1.1655901315529902), ('Utp18', 0.5...</td>\n",
       "      <td>472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M4579</th>\n",
       "      <td>0.058929</td>\n",
       "      <td>3.228891</td>\n",
       "      <td>0.000998</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>motif similar to swissregulon__mm__ARNT_ARNT2_...</td>\n",
       "      <td>frozenset({'activating', 'weight&gt;75.0%', 'mm10...</td>\n",
       "      <td>[('Rnf115', 1.1655901315529902), ('Map3k12', 0...</td>\n",
       "      <td>1520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">Zfp729a</th>\n",
       "      <th>dbcorrdb__YY1__ENCSR000BNT_1__m1</th>\n",
       "      <td>0.062308</td>\n",
       "      <td>3.295964</td>\n",
       "      <td>0.000241</td>\n",
       "      <td>0.329135</td>\n",
       "      <td>motif similar to transfac_pro__M06259 ('V$ZNF9...</td>\n",
       "      <td>frozenset({'activating', 'mm10__refseq-r80__50...</td>\n",
       "      <td>[('Hexim2', 0.3951984944559493), ('Celf1', 0.5...</td>\n",
       "      <td>1370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbcorrdb__YY1__ENCSR000EWF_1__m1</th>\n",
       "      <td>0.061692</td>\n",
       "      <td>3.234288</td>\n",
       "      <td>0.000824</td>\n",
       "      <td>0.329135</td>\n",
       "      <td>motif similar to transfac_pro__M06259 ('V$ZNF9...</td>\n",
       "      <td>frozenset({'activating', 'mm10__refseq-r80__50...</td>\n",
       "      <td>[('Hexim2', 0.3951984944559493), ('Celf1', 0.5...</td>\n",
       "      <td>1741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbcorrdb__YY1__ENCSR000BNX_1__m1</th>\n",
       "      <td>0.060918</td>\n",
       "      <td>3.156835</td>\n",
       "      <td>0.000332</td>\n",
       "      <td>0.329135</td>\n",
       "      <td>motif similar to transfac_pro__M06259 ('V$ZNF9...</td>\n",
       "      <td>frozenset({'activating', 'mm10__refseq-r80__50...</td>\n",
       "      <td>[('Hexim2', 0.3951984944559493), ('Celf1', 0.5...</td>\n",
       "      <td>1475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dbcorrdb__YY1__ENCSR000BKD_1__m1</th>\n",
       "      <td>0.061782</td>\n",
       "      <td>3.243253</td>\n",
       "      <td>0.000627</td>\n",
       "      <td>0.329135</td>\n",
       "      <td>motif similar to transfac_pro__M06259 ('V$ZNF9...</td>\n",
       "      <td>frozenset({'activating', 'mm10__refseq-r80__50...</td>\n",
       "      <td>[('Hexim2', 0.3951984944559493), ('Celf1', 0.5...</td>\n",
       "      <td>1313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Zmiz1</th>\n",
       "      <th>transfac_pro__M07520</th>\n",
       "      <td>0.055091</td>\n",
       "      <td>3.013477</td>\n",
       "      <td>0.000323</td>\n",
       "      <td>0.975746</td>\n",
       "      <td>gene is orthologous to ENSG00000108175 in H. s...</td>\n",
       "      <td>frozenset({'activating', 'mm10__refseq-r80__50...</td>\n",
       "      <td>[('Zmiz1', 1.0), ('Tox', 0.4034296260345378), ...</td>\n",
       "      <td>836</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26486 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Enrichment            \\\n",
       "                                                       AUC       NES   \n",
       "TF      MotifID                                                        \n",
       "Ar      transfac_pro__M08908                      0.073686  3.037615   \n",
       "Arnt    jaspar__MA0603.1                          0.057200  3.073739   \n",
       "        hocomoco__USF1_MOUSE.H11MO.0.A            0.066981  3.951676   \n",
       "        taipale_cyt_meth__ARNTL_RTCACGTGMN_eDBD   0.057509  3.101489   \n",
       "        cisbp__M4579                              0.058929  3.228891   \n",
       "...                                                    ...       ...   \n",
       "Zfp729a dbcorrdb__YY1__ENCSR000BNT_1__m1          0.062308  3.295964   \n",
       "        dbcorrdb__YY1__ENCSR000EWF_1__m1          0.061692  3.234288   \n",
       "        dbcorrdb__YY1__ENCSR000BNX_1__m1          0.060918  3.156835   \n",
       "        dbcorrdb__YY1__ENCSR000BKD_1__m1          0.061782  3.243253   \n",
       "Zmiz1   transfac_pro__M07520                      0.055091  3.013477   \n",
       "\n",
       "                                                                       \\\n",
       "                                                MotifSimilarityQvalue   \n",
       "TF      MotifID                                                         \n",
       "Ar      transfac_pro__M08908                                 0.000000   \n",
       "Arnt    jaspar__MA0603.1                                     0.000799   \n",
       "        hocomoco__USF1_MOUSE.H11MO.0.A                       0.000675   \n",
       "        taipale_cyt_meth__ARNTL_RTCACGTGMN_eDBD              0.000479   \n",
       "        cisbp__M4579                                         0.000998   \n",
       "...                                                               ...   \n",
       "Zfp729a dbcorrdb__YY1__ENCSR000BNT_1__m1                     0.000241   \n",
       "        dbcorrdb__YY1__ENCSR000EWF_1__m1                     0.000824   \n",
       "        dbcorrdb__YY1__ENCSR000BNX_1__m1                     0.000332   \n",
       "        dbcorrdb__YY1__ENCSR000BKD_1__m1                     0.000627   \n",
       "Zmiz1   transfac_pro__M07520                                 0.000323   \n",
       "\n",
       "                                                                     \\\n",
       "                                                OrthologousIdentity   \n",
       "TF      MotifID                                                       \n",
       "Ar      transfac_pro__M08908                               0.877642   \n",
       "Arnt    jaspar__MA0603.1                                   1.000000   \n",
       "        hocomoco__USF1_MOUSE.H11MO.0.A                     1.000000   \n",
       "        taipale_cyt_meth__ARNTL_RTCACGTGMN_eDBD            0.389381   \n",
       "        cisbp__M4579                                       1.000000   \n",
       "...                                                             ...   \n",
       "Zfp729a dbcorrdb__YY1__ENCSR000BNT_1__m1                   0.329135   \n",
       "        dbcorrdb__YY1__ENCSR000EWF_1__m1                   0.329135   \n",
       "        dbcorrdb__YY1__ENCSR000BNX_1__m1                   0.329135   \n",
       "        dbcorrdb__YY1__ENCSR000BKD_1__m1                   0.329135   \n",
       "Zmiz1   transfac_pro__M07520                               0.975746   \n",
       "\n",
       "                                                                                                    \\\n",
       "                                                                                        Annotation   \n",
       "TF      MotifID                                                                                      \n",
       "Ar      transfac_pro__M08908                     gene is orthologous to ENSG00000169083 in H. s...   \n",
       "Arnt    jaspar__MA0603.1                         gene is annotated for similar motif transfac_p...   \n",
       "        hocomoco__USF1_MOUSE.H11MO.0.A           gene is annotated for similar motif swissregul...   \n",
       "        taipale_cyt_meth__ARNTL_RTCACGTGMN_eDBD  gene is orthologous to FBgn0264075 in D. melan...   \n",
       "        cisbp__M4579                             motif similar to swissregulon__mm__ARNT_ARNT2_...   \n",
       "...                                                                                            ...   \n",
       "Zfp729a dbcorrdb__YY1__ENCSR000BNT_1__m1         motif similar to transfac_pro__M06259 ('V$ZNF9...   \n",
       "        dbcorrdb__YY1__ENCSR000EWF_1__m1         motif similar to transfac_pro__M06259 ('V$ZNF9...   \n",
       "        dbcorrdb__YY1__ENCSR000BNX_1__m1         motif similar to transfac_pro__M06259 ('V$ZNF9...   \n",
       "        dbcorrdb__YY1__ENCSR000BKD_1__m1         motif similar to transfac_pro__M06259 ('V$ZNF9...   \n",
       "Zmiz1   transfac_pro__M07520                     gene is orthologous to ENSG00000108175 in H. s...   \n",
       "\n",
       "                                                                                                    \\\n",
       "                                                                                           Context   \n",
       "TF      MotifID                                                                                      \n",
       "Ar      transfac_pro__M08908                     frozenset({'activating', 'weight>75.0%', 'mm10...   \n",
       "Arnt    jaspar__MA0603.1                         frozenset({'activating', 'weight>75.0%', 'mm10...   \n",
       "        hocomoco__USF1_MOUSE.H11MO.0.A           frozenset({'activating', 'weight>75.0%', 'mm10...   \n",
       "        taipale_cyt_meth__ARNTL_RTCACGTGMN_eDBD  frozenset({'activating', 'weight>75.0%', 'mm10...   \n",
       "        cisbp__M4579                             frozenset({'activating', 'weight>75.0%', 'mm10...   \n",
       "...                                                                                            ...   \n",
       "Zfp729a dbcorrdb__YY1__ENCSR000BNT_1__m1         frozenset({'activating', 'mm10__refseq-r80__50...   \n",
       "        dbcorrdb__YY1__ENCSR000EWF_1__m1         frozenset({'activating', 'mm10__refseq-r80__50...   \n",
       "        dbcorrdb__YY1__ENCSR000BNX_1__m1         frozenset({'activating', 'mm10__refseq-r80__50...   \n",
       "        dbcorrdb__YY1__ENCSR000BKD_1__m1         frozenset({'activating', 'mm10__refseq-r80__50...   \n",
       "Zmiz1   transfac_pro__M07520                     frozenset({'activating', 'mm10__refseq-r80__50...   \n",
       "\n",
       "                                                                                                    \\\n",
       "                                                                                       TargetGenes   \n",
       "TF      MotifID                                                                                      \n",
       "Ar      transfac_pro__M08908                     [('Ar', 1.0), ('Rdh12', 0.4101501149570101), (...   \n",
       "Arnt    jaspar__MA0603.1                         [('Rnf115', 1.1655901315529902), ('Klf6', 0.93...   \n",
       "        hocomoco__USF1_MOUSE.H11MO.0.A           [('Rnf115', 1.1655901315529902), ('Map3k12', 0...   \n",
       "        taipale_cyt_meth__ARNTL_RTCACGTGMN_eDBD  [('Rnf115', 1.1655901315529902), ('Utp18', 0.5...   \n",
       "        cisbp__M4579                             [('Rnf115', 1.1655901315529902), ('Map3k12', 0...   \n",
       "...                                                                                            ...   \n",
       "Zfp729a dbcorrdb__YY1__ENCSR000BNT_1__m1         [('Hexim2', 0.3951984944559493), ('Celf1', 0.5...   \n",
       "        dbcorrdb__YY1__ENCSR000EWF_1__m1         [('Hexim2', 0.3951984944559493), ('Celf1', 0.5...   \n",
       "        dbcorrdb__YY1__ENCSR000BNX_1__m1         [('Hexim2', 0.3951984944559493), ('Celf1', 0.5...   \n",
       "        dbcorrdb__YY1__ENCSR000BKD_1__m1         [('Hexim2', 0.3951984944559493), ('Celf1', 0.5...   \n",
       "Zmiz1   transfac_pro__M07520                     [('Zmiz1', 1.0), ('Tox', 0.4034296260345378), ...   \n",
       "\n",
       "                                                           \n",
       "                                                RankAtMax  \n",
       "TF      MotifID                                            \n",
       "Ar      transfac_pro__M08908                         1359  \n",
       "Arnt    jaspar__MA0603.1                              471  \n",
       "        hocomoco__USF1_MOUSE.H11MO.0.A               1118  \n",
       "        taipale_cyt_meth__ARNTL_RTCACGTGMN_eDBD       472  \n",
       "        cisbp__M4579                                 1520  \n",
       "...                                                   ...  \n",
       "Zfp729a dbcorrdb__YY1__ENCSR000BNT_1__m1             1370  \n",
       "        dbcorrdb__YY1__ENCSR000EWF_1__m1             1741  \n",
       "        dbcorrdb__YY1__ENCSR000BNX_1__m1             1475  \n",
       "        dbcorrdb__YY1__ENCSR000BKD_1__m1             1313  \n",
       "Zmiz1   transfac_pro__M07520                          836  \n",
       "\n",
       "[26486 rows x 8 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regulons_fl"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed0e605c",
   "metadata": {},
   "source": [
    "## Rorc(+)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "7d8898f2",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-22T17:00:30.515908Z",
     "start_time": "2022-11-22T17:00:30.485932Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th colspan=\"8\" halign=\"left\">Enrichment</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>NES</th>\n",
       "      <th>MotifSimilarityQvalue</th>\n",
       "      <th>OrthologousIdentity</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Context</th>\n",
       "      <th>TargetGenes</th>\n",
       "      <th>RankAtMax</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TF</th>\n",
       "      <th>MotifID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">Rorc</th>\n",
       "      <th>swissregulon__hs__RORA.p2</th>\n",
       "      <td>0.076715</td>\n",
       "      <td>3.492629</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>motif similar to hocomoco__RORG_MOUSE.H11MO.0....</td>\n",
       "      <td>frozenset({'activating', 'top5perTarget', 'mm1...</td>\n",
       "      <td>[('Lmo4', 15.397134524481867), ('S100a4', 16.3...</td>\n",
       "      <td>404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__NR1D2_MOUSE.H11MO.0.A</th>\n",
       "      <td>0.135071</td>\n",
       "      <td>3.291875</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>motif similar to hocomoco__RORG_MOUSE.H11MO.0....</td>\n",
       "      <td>frozenset({'activating', 'mm10__refseq-r80__50...</td>\n",
       "      <td>[('Rorc', 1.0), ('Cd82', 30.98627131405585), (...</td>\n",
       "      <td>822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M6455</th>\n",
       "      <td>0.131941</td>\n",
       "      <td>3.172258</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.891473</td>\n",
       "      <td>gene is orthologous to ENSG00000143365 in H. s...</td>\n",
       "      <td>frozenset({'activating', 'mm10__refseq-r80__50...</td>\n",
       "      <td>[('Lmo4', 15.397134524481867), ('S100a4', 16.3...</td>\n",
       "      <td>618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>swissregulon__hs__RORA.p2</th>\n",
       "      <td>0.130556</td>\n",
       "      <td>3.119302</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>motif similar to hocomoco__RORG_MOUSE.H11MO.0....</td>\n",
       "      <td>frozenset({'activating', 'mm10__refseq-r80__50...</td>\n",
       "      <td>[('Lmo4', 15.397134524481867), ('S100a4', 16.3...</td>\n",
       "      <td>1057</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     Enrichment            \\\n",
       "                                            AUC       NES   \n",
       "TF   MotifID                                                \n",
       "Rorc swissregulon__hs__RORA.p2         0.076715  3.492629   \n",
       "     hocomoco__NR1D2_MOUSE.H11MO.0.A   0.135071  3.291875   \n",
       "     cisbp__M6455                      0.131941  3.172258   \n",
       "     swissregulon__hs__RORA.p2         0.130556  3.119302   \n",
       "\n",
       "                                                            \\\n",
       "                                     MotifSimilarityQvalue   \n",
       "TF   MotifID                                                 \n",
       "Rorc swissregulon__hs__RORA.p2                    0.000067   \n",
       "     hocomoco__NR1D2_MOUSE.H11MO.0.A              0.000032   \n",
       "     cisbp__M6455                                 0.000000   \n",
       "     swissregulon__hs__RORA.p2                    0.000067   \n",
       "\n",
       "                                                          \\\n",
       "                                     OrthologousIdentity   \n",
       "TF   MotifID                                               \n",
       "Rorc swissregulon__hs__RORA.p2                  1.000000   \n",
       "     hocomoco__NR1D2_MOUSE.H11MO.0.A            1.000000   \n",
       "     cisbp__M6455                               0.891473   \n",
       "     swissregulon__hs__RORA.p2                  1.000000   \n",
       "\n",
       "                                                                                         \\\n",
       "                                                                             Annotation   \n",
       "TF   MotifID                                                                              \n",
       "Rorc swissregulon__hs__RORA.p2        motif similar to hocomoco__RORG_MOUSE.H11MO.0....   \n",
       "     hocomoco__NR1D2_MOUSE.H11MO.0.A  motif similar to hocomoco__RORG_MOUSE.H11MO.0....   \n",
       "     cisbp__M6455                     gene is orthologous to ENSG00000143365 in H. s...   \n",
       "     swissregulon__hs__RORA.p2        motif similar to hocomoco__RORG_MOUSE.H11MO.0....   \n",
       "\n",
       "                                                                                         \\\n",
       "                                                                                Context   \n",
       "TF   MotifID                                                                              \n",
       "Rorc swissregulon__hs__RORA.p2        frozenset({'activating', 'top5perTarget', 'mm1...   \n",
       "     hocomoco__NR1D2_MOUSE.H11MO.0.A  frozenset({'activating', 'mm10__refseq-r80__50...   \n",
       "     cisbp__M6455                     frozenset({'activating', 'mm10__refseq-r80__50...   \n",
       "     swissregulon__hs__RORA.p2        frozenset({'activating', 'mm10__refseq-r80__50...   \n",
       "\n",
       "                                                                                         \\\n",
       "                                                                            TargetGenes   \n",
       "TF   MotifID                                                                              \n",
       "Rorc swissregulon__hs__RORA.p2        [('Lmo4', 15.397134524481867), ('S100a4', 16.3...   \n",
       "     hocomoco__NR1D2_MOUSE.H11MO.0.A  [('Rorc', 1.0), ('Cd82', 30.98627131405585), (...   \n",
       "     cisbp__M6455                     [('Lmo4', 15.397134524481867), ('S100a4', 16.3...   \n",
       "     swissregulon__hs__RORA.p2        [('Lmo4', 15.397134524481867), ('S100a4', 16.3...   \n",
       "\n",
       "                                                \n",
       "                                     RankAtMax  \n",
       "TF   MotifID                                    \n",
       "Rorc swissregulon__hs__RORA.p2             404  \n",
       "     hocomoco__NR1D2_MOUSE.H11MO.0.A       822  \n",
       "     cisbp__M6455                          618  \n",
       "     swissregulon__hs__RORA.p2            1057  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "selected_motifs = ['Rorc']\n",
    "df_motifs_sel = regulons_fl.iloc[ [ True if x in selected_motifs else False for x in regulons_fl.index.get_level_values('TF') ] ,:]\n",
    "df_motifs_sel = df_motifs_sel.sort_values([('Enrichment','NES')], ascending=False)\n",
    "df_motifs_sel\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "dba1d11e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-22T17:00:55.837255Z",
     "start_time": "2022-11-22T17:00:52.363970Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th colspan=\"9\" halign=\"left\">Enrichment</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>AUC</th>\n",
       "      <th>NES</th>\n",
       "      <th>MotifSimilarityQvalue</th>\n",
       "      <th>OrthologousIdentity</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Context</th>\n",
       "      <th>TargetGenes</th>\n",
       "      <th>RankAtMax</th>\n",
       "      <th>MotifLogo</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TF</th>\n",
       "      <th>MotifID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">Rorc</th>\n",
       "      <th>swissregulon__hs__RORA.p2</th>\n",
       "      <td>0.076715</td>\n",
       "      <td>3.492629</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>motif similar to hocomoco__RORG_MOUSE.H11MO.0.B ('RORG_MOUSE'; q-value = 6.7e-05) which is directly annotated</td>\n",
       "      <td>frozenset({'activating', 'top5perTarget', 'mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr'})</td>\n",
       "      <td>[('Lmo4', 15.397134524481867), ('S100a4', 16.33933560660286), ('Il17re', 7.325174159511872), ('Il17f', 2.946725618556415), ('Cd160', 8.424694972597475), ('Ppt2', 12.980209356981472), ('Tjp3', 0.9077859684726372), ('Cd82', 30.98627131405585), ('Mif4gd', 4.758108965634077), ('Sla', 11.331077763622652), ('Gnb5', 0.759108126701642), ('Rorc', 1.0), ('Unc119', 16.654630295757677)]</td>\n",
       "      <td>404</td>\n",
       "      <td><img src=\"http://motifcollections.aertslab.org/v9/logos/swissregulon__hs__RORA.p2.png\" style=\"max-height:124px;\"></img></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hocomoco__NR1D2_MOUSE.H11MO.0.A</th>\n",
       "      <td>0.135071</td>\n",
       "      <td>3.291875</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>motif similar to hocomoco__RORG_MOUSE.H11MO.0.B ('RORG_MOUSE'; q-value = 3.18e-05) which is directly annotated</td>\n",
       "      <td>frozenset({'activating', 'mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr', 'top50'})</td>\n",
       "      <td>[('Rorc', 1.0), ('Cd82', 30.98627131405585), ('S100a4', 16.33933560660286), ('Tox', 25.754493315118022), ('Lmo4', 15.397134524481867), ('Tpm2', 10.42797596053166), ('Sla', 11.331077763622652), ('Vasp', 9.645696262882351), ('Ppt2', 12.980209356981472), ('Ltb', 18.55908932493929)]</td>\n",
       "      <td>822</td>\n",
       "      <td><img src=\"http://motifcollections.aertslab.org/v9/logos/hocomoco__NR1D2_MOUSE.H11MO.0.A.png\" style=\"max-height:124px;\"></img></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cisbp__M6455</th>\n",
       "      <td>0.131941</td>\n",
       "      <td>3.172258</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.891473</td>\n",
       "      <td>gene is orthologous to ENSG00000143365 in H. sapiens (identity = 89%) which is directly annotated for motif</td>\n",
       "      <td>frozenset({'activating', 'mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr', 'top50'})</td>\n",
       "      <td>[('Lmo4', 15.397134524481867), ('S100a4', 16.33933560660286), ('Rorc', 1.0), ('Cd82', 30.98627131405585), ('Ppt2', 12.980209356981472), ('Anxa6', 12.653389443445896), ('Sla', 11.331077763622652), ('Cldnd1', 11.433269822989883), ('Zap70', 14.446255708950432)]</td>\n",
       "      <td>618</td>\n",
       "      <td><img src=\"http://motifcollections.aertslab.org/v9/logos/cisbp__M6455.png\" style=\"max-height:124px;\"></img></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>swissregulon__hs__RORA.p2</th>\n",
       "      <td>0.130556</td>\n",
       "      <td>3.119302</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>motif similar to hocomoco__RORG_MOUSE.H11MO.0.B ('RORG_MOUSE'; q-value = 6.7e-05) which is directly annotated</td>\n",
       "      <td>frozenset({'activating', 'mm10__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr', 'top50'})</td>\n",
       "      <td>[('Lmo4', 15.397134524481867), ('S100a4', 16.33933560660286), ('Ppt2', 12.980209356981472), ('Cd82', 30.98627131405585), ('Sla', 11.331077763622652), ('Rorc', 1.0), ('Unc119', 16.654630295757677), ('Anxa6', 12.653389443445896), ('Chad', 20.313881512509813), ('Zap70', 14.446255708950432), ('Cldnd1', 11.433269822989883)]</td>\n",
       "      <td>1057</td>\n",
       "      <td><img src=\"http://motifcollections.aertslab.org/v9/logos/swissregulon__hs__RORA.p2.png\" style=\"max-height:124px;\"></img></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from pyscenic.utils import load_motifs\n",
    "import operator as op\n",
    "from IPython.display import HTML, display\n",
    "\n",
    "BASE_URL = \"http://motifcollections.aertslab.org/v9/logos/\"\n",
    "COLUMN_NAME_LOGO = \"MotifLogo\"\n",
    "COLUMN_NAME_MOTIF_ID = \"MotifID\"\n",
    "COLUMN_NAME_TARGETS = \"TargetGenes\"\n",
    "\n",
    "def display_logos(df: pd.DataFrame, top_target_genes: int = 3, base_url: str = BASE_URL):\n",
    "    \"\"\"\n",
    "    :param df:\n",
    "    :param base_url:\n",
    "    \"\"\"\n",
    "    # Make sure the original dataframe is not altered.\n",
    "    df = df.copy()\n",
    "    \n",
    "    # Add column with URLs to sequence logo.\n",
    "    def create_url(motif_id):\n",
    "        return '<img src=\"{}{}.png\" style=\"max-height:124px;\"></img>'.format(base_url, motif_id)\n",
    "    df[(\"Enrichment\", COLUMN_NAME_LOGO)] = list(map(create_url, df.index.get_level_values(COLUMN_NAME_MOTIF_ID)))\n",
    "    \n",
    "    # Truncate TargetGenes.\n",
    "    def truncate(col_val):\n",
    "        return sorted(col_val, key=op.itemgetter(1))[:top_target_genes]\n",
    "    \n",
    "    #df[(\"Enrichment\", COLUMN_NAME_TARGETS)] = list(map(truncate, df[(\"Enrichment\", COLUMN_NAME_TARGETS)]))\n",
    "    \n",
    "    MAX_COL_WIDTH = pd.get_option('display.max_colwidth')\n",
    "    pd.set_option('display.max_colwidth', 200)\n",
    "    display(HTML(df.head().to_html(escape=False)))\n",
    "    pd.set_option('display.max_colwidth', MAX_COL_WIDTH)\n",
    "    \n",
    "display_logos(df_motifs_sel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "baacda5b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-23T08:33:45.289130Z",
     "start_time": "2022-11-23T08:33:45.215411Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>target</th>\n",
       "      <th>importance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Cd82</td>\n",
       "      <td>30.986271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Tox</td>\n",
       "      <td>25.754493</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Chad</td>\n",
       "      <td>20.313882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Ltb</td>\n",
       "      <td>18.559089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Unc119</td>\n",
       "      <td>16.654630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S100a4</td>\n",
       "      <td>16.339336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Lmo4</td>\n",
       "      <td>15.397135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Zap70</td>\n",
       "      <td>14.446256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Ppt2</td>\n",
       "      <td>12.980209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Anxa6</td>\n",
       "      <td>12.653389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Cldnd1</td>\n",
       "      <td>11.433270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Sla</td>\n",
       "      <td>11.331078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Tpm2</td>\n",
       "      <td>10.427976</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Vasp</td>\n",
       "      <td>9.645696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Cd160</td>\n",
       "      <td>8.424695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Il17re</td>\n",
       "      <td>7.325174</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Mif4gd</td>\n",
       "      <td>4.758109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Il17f</td>\n",
       "      <td>2.946726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Rorc</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Tjp3</td>\n",
       "      <td>0.907786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Gnb5</td>\n",
       "      <td>0.759108</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    target  importance\n",
       "3     Cd82   30.986271\n",
       "11     Tox   25.754493\n",
       "10    Chad   20.313882\n",
       "14     Ltb   18.559089\n",
       "9   Unc119   16.654630\n",
       "1   S100a4   16.339336\n",
       "0     Lmo4   15.397135\n",
       "8    Zap70   14.446256\n",
       "4     Ppt2   12.980209\n",
       "5    Anxa6   12.653389\n",
       "7   Cldnd1   11.433270\n",
       "6      Sla   11.331078\n",
       "12    Tpm2   10.427976\n",
       "13    Vasp    9.645696\n",
       "17   Cd160    8.424695\n",
       "15  Il17re    7.325174\n",
       "19  Mif4gd    4.758109\n",
       "16   Il17f    2.946726\n",
       "2     Rorc    1.000000\n",
       "18    Tjp3    0.907786\n",
       "20    Gnb5    0.759108"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "targets = regulons_fl.loc['Rorc',['Enrichment']].droplevel(0, axis=1)['TargetGenes']\n",
    "targets = [eval(tg) for tg in targets]\n",
    "import numpy as np\n",
    "res ={}\n",
    "for target in targets:\n",
    "    for tup in target:\n",
    "        if tup[0] in res.keys():\n",
    "            res[tup[0]].append( float(tup[1]))\n",
    "        else:\n",
    "            res[tup[0]] = [ tup[1] ]\n",
    "\n",
    "\n",
    "for k,v in res.items():\n",
    "    res[k] = np.mean(v) \n",
    "targets = pd.DataFrame({\"target\": res.keys(), \"importance\": res.values() })\\\n",
    "         .sort_values('importance',ascending=False)\n",
    "\n",
    "targets.to_csv('220517-SCENICprediction-FL-Targets-of-Rorc.csv')\n",
    "targets"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
